// ant-design-vue/lib/style/compact-item-vertical.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.genCompactItemVerticalStyle = genCompactItemVerticalStyle;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
function compactItemVerticalBorder(token, parentCls) {
  var _ref;
  return _ref = {}, (0, _defineProperty2.default)(_ref, "&-item:not(".concat(parentCls, "-last-item)"), {
    marginBottom: -token.lineWidth
  }), (0, _defineProperty2.default)(_ref, '&-item', {
    '&:hover,&:focus,&:active': {
      zIndex: 2
    },
    '&[disabled]': {
      zIndex: 0
    }
  }), _ref;
}
function compactItemBorderVerticalRadius(prefixCls, parentCls) {
  var _ref2;
  return _ref2 = {}, (0, _defineProperty2.default)(_ref2, "&-item:not(".concat(parentCls, "-first-item):not(").concat(parentCls, "-last-item)"), {
    borderRadius: 0
  }), (0, _defineProperty2.default)(_ref2, "&-item".concat(parentCls, "-first-item:not(").concat(parentCls, "-last-item)"), (0, _defineProperty2.default)({}, "&, &".concat(prefixCls, "-sm, &").concat(prefixCls, "-lg"), {
    borderEndEndRadius: 0,
    borderEndStartRadius: 0
  })), (0, _defineProperty2.default)(_ref2, "&-item".concat(parentCls, "-last-item:not(").concat(parentCls, "-first-item)"), (0, _defineProperty2.default)({}, "&, &".concat(prefixCls, "-sm, &").concat(prefixCls, "-lg"), {
    borderStartStartRadius: 0,
    borderStartEndRadius: 0
  })), _ref2;
}
function genCompactItemVerticalStyle(token) {
  var compactCls = "".concat(token.componentCls, "-compact-vertical");
  return (0, _defineProperty2.default)({}, compactCls, (0, _extends2.default)((0, _extends2.default)({}, compactItemVerticalBorder(token, compactCls)), compactItemBorderVerticalRadius(token.componentCls, compactCls)));
}
