// ant-design-vue/lib/upload/style/motion.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _cssinjs = require("../../_util/cssinjs");
var uploadAnimateInlineIn = new _cssinjs.Keyframes('uploadAnimateInlineIn', {
  from: {
    width: 0,
    height: 0,
    margin: 0,
    padding: 0,
    opacity: 0
  }
});
var uploadAnimateInlineOut = new _cssinjs.Keyframes('uploadAnimateInlineOut', {
  to: {
    width: 0,
    height: 0,
    margin: 0,
    padding: 0,
    opacity: 0
  }
});
// =========================== Motion ===========================
var genMotionStyle = function genMotionStyle(token) {
  var _$concat;
  var componentCls = token.componentCls;
  var inlineCls = "".concat(componentCls, "-animate-inline");
  return [(0, _defineProperty2.default)({}, "".concat(componentCls, "-wrapper"), (_$concat = {}, (0, _defineProperty2.default)(_$concat, "".concat(inlineCls, "-appear, ").concat(inlineCls, "-enter, ").concat(inlineCls, "-leave"), {
    animationDuration: token.motionDurationSlow,
    animationTimingFunction: token.motionEaseInOutCirc,
    animationFillMode: 'forwards'
  }), (0, _defineProperty2.default)(_$concat, "".concat(inlineCls, "-appear, ").concat(inlineCls, "-enter"), {
    animationName: uploadAnimateInlineIn
  }), (0, _defineProperty2.default)(_$concat, "".concat(inlineCls, "-leave"), {
    animationName: uploadAnimateInlineOut
  }), _$concat)), uploadAnimateInlineIn, uploadAnimateInlineOut];
};
var _default = genMotionStyle;
exports.default = _default;
