// ant-design-vue/lib/calendar/style/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.genCalendarStyles = exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _extends3 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _style = require("../../style");
var _style2 = require("../../date-picker/style");
var _style3 = require("../../input/style");
var _internal = require("../../theme/internal");
var genCalendarStyles = function genCalendarStyles(token) {
  var _$concat, _$concat$concat, _$concat2, _$concat4, _$concat$concat3, _$concat5, _ref;
  var calendarCls = token.calendarCls,
    componentCls = token.componentCls,
    calendarFullBg = token.calendarFullBg,
    calendarFullPanelBg = token.calendarFullPanelBg,
    calendarItemActiveBg = token.calendarItemActiveBg;
  return _ref = {}, (0, _defineProperty2.default)(_ref, calendarCls, (0, _extends3.default)((0, _extends3.default)((0, _extends3.default)({}, (0, _style2.genPanelStyle)(token)), (0, _style.resetComponent)(token)), (0, _defineProperty2.default)({
    background: calendarFullBg,
    '&-rtl': {
      direction: 'rtl'
    }
  }, "".concat(calendarCls, "-header"), (_$concat = {
    display: 'flex',
    justifyContent: 'flex-end',
    padding: "".concat(token.paddingSM, "px 0")
  }, (0, _defineProperty2.default)(_$concat, "".concat(calendarCls, "-year-select"), {
    minWidth: token.yearControlWidth
  }), (0, _defineProperty2.default)(_$concat, "".concat(calendarCls, "-month-select"), {
    minWidth: token.monthControlWidth,
    marginInlineStart: token.marginXS
  }), (0, _defineProperty2.default)(_$concat, "".concat(calendarCls, "-mode-switch"), {
    marginInlineStart: token.marginXS
  }), _$concat)))), (0, _defineProperty2.default)(_ref, "".concat(calendarCls, " ").concat(componentCls, "-panel"), (_$concat$concat = {
    background: calendarFullPanelBg,
    border: 0,
    borderTop: "".concat(token.lineWidth, "px ").concat(token.lineType, " ").concat(token.colorSplit),
    borderRadius: 0
  }, (0, _defineProperty2.default)(_$concat$concat, "".concat(componentCls, "-month-panel, ").concat(componentCls, "-date-panel"), {
    width: 'auto'
  }), (0, _defineProperty2.default)(_$concat$concat, "".concat(componentCls, "-body"), {
    padding: "".concat(token.paddingXS, "px 0")
  }), (0, _defineProperty2.default)(_$concat$concat, "".concat(componentCls, "-content"), {
    width: '100%'
  }), _$concat$concat)), (0, _defineProperty2.default)(_ref, "".concat(calendarCls, "-mini"), (_$concat2 = {
    borderRadius: token.borderRadiusLG
  }, (0, _defineProperty2.default)(_$concat2, "".concat(calendarCls, "-header"), {
    paddingInlineEnd: token.paddingXS,
    paddingInlineStart: token.paddingXS
  }), (0, _defineProperty2.default)(_$concat2, "".concat(componentCls, "-panel"), {
    borderRadius: "0 0 ".concat(token.borderRadiusLG, "px ").concat(token.borderRadiusLG, "px")
  }), (0, _defineProperty2.default)(_$concat2, "".concat(componentCls, "-content"), {
    height: token.miniContentHeight,
    th: {
      height: 'auto',
      padding: 0,
      lineHeight: "".concat(token.weekHeight, "px")
    }
  }), (0, _defineProperty2.default)(_$concat2, "".concat(componentCls, "-cell::before"), {
    pointerEvents: 'none'
  }), _$concat2)), (0, _defineProperty2.default)(_ref, "".concat(calendarCls).concat(calendarCls, "-full"), (_$concat$concat3 = {}, (0, _defineProperty2.default)(_$concat$concat3, "".concat(componentCls, "-panel"), (0, _defineProperty2.default)({
    display: 'block',
    width: '100%',
    textAlign: 'end',
    background: calendarFullBg,
    border: 0
  }, "".concat(componentCls, "-body"), {
    'th, td': {
      padding: 0
    },
    th: {
      height: 'auto',
      paddingInlineEnd: token.paddingSM,
      paddingBottom: token.paddingXXS,
      lineHeight: "".concat(token.weekHeight, "px")
    }
  })), (0, _defineProperty2.default)(_$concat$concat3, "".concat(componentCls, "-cell"), (_$concat4 = {
    '&::before': {
      display: 'none'
    },
    '&:hover': (0, _defineProperty2.default)({}, "".concat(calendarCls, "-date"), {
      background: token.controlItemBgHover
    })
  }, (0, _defineProperty2.default)(_$concat4, "".concat(calendarCls, "-date-today::before"), {
    display: 'none'
  }), (0, _defineProperty2.default)(_$concat4, "&-in-view".concat(componentCls, "-cell-selected"), (0, _defineProperty2.default)({}, "".concat(calendarCls, "-date, ").concat(calendarCls, "-date-today"), {
    background: calendarItemActiveBg
  })), (0, _defineProperty2.default)(_$concat4, '&-selected, &-selected:hover', (0, _defineProperty2.default)({}, "".concat(calendarCls, "-date, ").concat(calendarCls, "-date-today"), (0, _defineProperty2.default)({}, "".concat(calendarCls, "-date-value"), {
    color: token.colorPrimary
  }))), _$concat4)), (0, _defineProperty2.default)(_$concat$concat3, "".concat(calendarCls, "-date"), {
    display: 'block',
    width: 'auto',
    height: 'auto',
    margin: "0 ".concat(token.marginXS / 2, "px"),
    padding: "".concat(token.paddingXS / 2, "px ").concat(token.paddingXS, "px 0"),
    border: 0,
    borderTop: "".concat(token.lineWidthBold, "px ").concat(token.lineType, " ").concat(token.colorSplit),
    borderRadius: 0,
    transition: "background ".concat(token.motionDurationSlow),
    '&-value': {
      lineHeight: "".concat(token.dateValueHeight, "px"),
      transition: "color ".concat(token.motionDurationSlow)
    },
    '&-content': {
      position: 'static',
      width: 'auto',
      height: token.dateContentHeight,
      overflowY: 'auto',
      color: token.colorText,
      lineHeight: token.lineHeight,
      textAlign: 'start'
    },
    '&-today': (0, _defineProperty2.default)({
      borderColor: token.colorPrimary
    }, "".concat(calendarCls, "-date-value"), {
      color: token.colorText
    })
  }), _$concat$concat3)), (0, _defineProperty2.default)(_ref, "@media only screen and (max-width: ".concat(token.screenXS, "px) "), (0, _defineProperty2.default)({}, "".concat(calendarCls), (0, _defineProperty2.default)({}, "".concat(calendarCls, "-header"), (_$concat5 = {
    display: 'block'
  }, (0, _defineProperty2.default)(_$concat5, "".concat(calendarCls, "-year-select"), {
    width: '50%'
  }), (0, _defineProperty2.default)(_$concat5, "".concat(calendarCls, "-month-select"), {
    width: "calc(50% - ".concat(token.paddingXS, "px)")
  }), (0, _defineProperty2.default)(_$concat5, "".concat(calendarCls, "-mode-switch"), {
    width: '100%',
    marginTop: token.marginXS,
    marginInlineStart: 0,
    '> label': {
      width: '50%',
      textAlign: 'center'
    }
  }), _$concat5)))), _ref;
};
exports.genCalendarStyles = genCalendarStyles;
var _default = (0, _internal.genComponentStyleHook)('Calendar', function (token) {
  var calendarCls = "".concat(token.componentCls, "-calendar");
  var calendarToken = (0, _internal.mergeToken)((0, _style3.initInputToken)(token), (0, _style2.initPickerPanelToken)(token), {
    calendarCls: calendarCls,
    pickerCellInnerCls: "".concat(token.componentCls, "-cell-inner"),
    calendarFullBg: token.colorBgContainer,
    calendarFullPanelBg: token.colorBgContainer,
    calendarItemActiveBg: token.controlItemBgActive,
    dateValueHeight: token.controlHeightSM,
    weekHeight: token.controlHeightSM * 0.75,
    dateContentHeight: (token.fontSizeSM * token.lineHeightSM + token.marginXS) * 3 + token.lineWidth * 2
  });
  return [genCalendarStyles(calendarToken)];
}, {
  yearControlWidth: 80,
  monthControlWidth: 70,
  miniContentHeight: 256
});
exports.default = _default;
