// ant-design-vue/lib/page-header/style/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _extends3 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _internal = require("../../theme/internal");
var _style = require("../../style");
var genPageHeaderStyle = function genPageHeaderStyle(token) {
  var _extra, _$concat2, _$concat4, _extends2;
  var componentCls = token.componentCls,
    antCls = token.antCls;
  return (0, _defineProperty2.default)({}, componentCls, (0, _extends3.default)((0, _extends3.default)({}, (0, _style.resetComponent)(token)), (_extends2 = {
    position: 'relative',
    padding: "".concat(token.pageHeaderPaddingVertical, "px ").concat(token.pageHeaderPadding, "px"),
    backgroundColor: token.colorBgContainer
  }, (0, _defineProperty2.default)(_extends2, "".concat(componentCls, "-ghost"), {
    backgroundColor: token.pageHeaderGhostBg
  }), (0, _defineProperty2.default)(_extends2, "&.has-footer", {
    paddingBottom: 0
  }), (0, _defineProperty2.default)(_extends2, "".concat(componentCls, "-back"), (0, _defineProperty2.default)({
    marginRight: token.marginMD,
    fontZize: token.fontSizeLG,
    lineHeight: 1
  }, "&-button", (0, _extends3.default)((0, _extends3.default)({}, (0, _style.operationUnit)(token)), {
    color: token.pageHeaderBackColor,
    cursor: 'pointer'
  }))), (0, _defineProperty2.default)(_extends2, "".concat(antCls, "-divider-vertical"), {
    height: '14px',
    margin: "0 ".concat(token.marginSM),
    verticalAlign: 'middle'
  }), (0, _defineProperty2.default)(_extends2, "".concat(antCls, "-breadcrumb + &-heading"), {
    marginTop: token.marginXS
  }), (0, _defineProperty2.default)(_extends2, "".concat(componentCls, "-heading"), (_$concat2 = {
    display: 'flex',
    justifyContent: 'space-between'
  }, (0, _defineProperty2.default)(_$concat2, "&-left", {
    display: 'flex',
    alignItems: 'center',
    margin: "".concat(token.marginXS / 2, "px 0"),
    overflow: 'hidden'
  }), (0, _defineProperty2.default)(_$concat2, "&-title", (0, _extends3.default)({
    marginRight: token.marginSM,
    marginBottom: 0,
    color: token.colorTextHeading,
    fontWeight: 600,
    fontSize: token.pageHeaderHeadingTitle,
    lineHeight: "".concat(token.controlHeight, "px")
  }, _style.textEllipsis)), (0, _defineProperty2.default)(_$concat2, "".concat(antCls, "-avatar"), {
    marginRight: token.marginSM
  }), (0, _defineProperty2.default)(_$concat2, "&-sub-title", (0, _extends3.default)({
    marginRight: token.marginSM,
    color: token.colorTextDescription,
    fontSize: token.pageHeaderHeadingSubTitle,
    lineHeight: token.lineHeight
  }, _style.textEllipsis)), (0, _defineProperty2.default)(_$concat2, "&-extra", (_extra = {
    margin: "".concat(token.marginXS / 2, "px 0"),
    whiteSpace: 'nowrap'
  }, (0, _defineProperty2.default)(_extra, "> *", {
    marginLeft: token.marginSM,
    whiteSpace: 'unset'
  }), (0, _defineProperty2.default)(_extra, "> *:first-child", {
    marginLeft: 0
  }), _extra)), _$concat2)), (0, _defineProperty2.default)(_extends2, "".concat(componentCls, "-content"), {
    paddingTop: token.pageHeaderContentPaddingVertical
  }), (0, _defineProperty2.default)(_extends2, "".concat(componentCls, "-footer"), (0, _defineProperty2.default)({
    marginTop: token.marginMD
  }, "".concat(antCls, "-tabs"), (_$concat4 = {}, (0, _defineProperty2.default)(_$concat4, "> ".concat(antCls, "-tabs-nav"), (0, _defineProperty2.default)({
    margin: 0
  }, "&::before", {
    border: 'none'
  })), (0, _defineProperty2.default)(_$concat4, "".concat(antCls, "-tabs-tab"), {
    paddingTop: token.paddingXS,
    paddingBottom: token.paddingXS,
    fontSize: token.pageHeaderTabFontSize
  }), _$concat4))), (0, _defineProperty2.default)(_extends2, "".concat(componentCls, "-compact ").concat(componentCls, "-heading"), {
    flexWrap: 'wrap'
  }), (0, _defineProperty2.default)(_extends2, "&".concat(token.componentCls, "-rtl"), {
    direction: 'rtl'
  }), _extends2)));
};
// ============================== Export ==============================
var _default = (0, _internal.genComponentStyleHook)('PageHeader', function (token) {
  var PageHeaderToken = (0, _internal.mergeToken)(token, {
    pageHeaderPadding: token.paddingLG,
    pageHeaderPaddingVertical: token.paddingMD,
    pageHeaderPaddingBreadcrumb: token.paddingSM,
    pageHeaderContentPaddingVertical: token.paddingSM,
    pageHeaderBackColor: token.colorTextBase,
    pageHeaderGhostBg: 'inherit',
    pageHeaderHeadingTitle: token.fontSizeHeading4,
    pageHeaderHeadingSubTitle: token.fontSize,
    pageHeaderTabFontSize: token.fontSizeLG
  });
  return [genPageHeaderStyle(PageHeaderToken)];
});
exports.default = _default;
