// ant-design-vue/es/vc-notification/HookNotification.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import _extends from "@babel/runtime/helpers/esm/extends";
import _typeof from "@babel/runtime/helpers/esm/typeof";
import { createVNode as _createVNode } from "vue";
import { watch, computed, defineComponent, ref, TransitionGroup } from 'vue';
import Notice from './Notice';
import { getTransitionGroupProps } from '../_util/transition';
import classNames from '../_util/classNames';
import Portal from '../_util/Portal';
var seed = 0;
var now = Date.now();
export function getUuid() {
  var id = seed;
  seed += 1;
  return "rcNotification_".concat(now, "_").concat(id);
}
var Notification = defineComponent({
  name: 'HookNotification',
  inheritAttrs: false,
  props: ['prefixCls', 'transitionName', 'animation', 'maxCount', 'closeIcon', 'hashId', 'remove', 'notices', 'getStyles', 'getClassName', 'onAllRemoved', 'getContainer'],
  setup: function setup(props, _ref) {
    var attrs = _ref.attrs,
      slots = _ref.slots;
    var hookRefs = new Map();
    var notices = computed(function () {
      return props.notices;
    });
    var transitionProps = computed(function () {
      var name = props.transitionName;
      if (!name && props.animation) {
        switch (_typeof(props.animation)) {
          case 'string':
            name = props.animation;
            break;
          case 'function':
            name = props.animation().name;
            break;
          case 'object':
            name = props.animation.name;
            break;
          default:
            name = "".concat(props.prefixCls, "-fade");
            break;
        }
      }
      return getTransitionGroupProps(name);
    });
    var remove = function remove(key) {
      return props.remove(key);
    };
    var placements = ref({});
    watch(notices, function () {
      var nextPlacements = {};
      // init placements with animation
      Object.keys(placements.value).forEach(function (placement) {
        nextPlacements[placement] = [];
      });
      props.notices.forEach(function (config) {
        var _config$notice$placem = config.notice.placement,
          placement = _config$notice$placem === void 0 ? 'topRight' : _config$notice$placem;
        if (placement) {
          nextPlacements[placement] = nextPlacements[placement] || [];
          nextPlacements[placement].push(config);
        }
      });
      placements.value = nextPlacements;
    });
    var placementList = computed(function () {
      return Object.keys(placements.value);
    });
    return function () {
      var _a;
      var prefixCls = props.prefixCls,
        _props$closeIcon = props.closeIcon,
        closeIcon = _props$closeIcon === void 0 ? (_a = slots.closeIcon) === null || _a === void 0 ? void 0 : _a.call(slots, {
          prefixCls: prefixCls
        }) : _props$closeIcon;
      var noticeNodes = placementList.value.map(function (placement) {
        var _className;
        var _a, _b;
        var noticesForPlacement = placements.value[placement];
        var classes = (_a = props.getClassName) === null || _a === void 0 ? void 0 : _a.call(props, placement);
        var styles = (_b = props.getStyles) === null || _b === void 0 ? void 0 : _b.call(props, placement);
        var noticeNodesForPlacement = noticesForPlacement.map(function (_ref2, index) {
          var notice = _ref2.notice,
            holderCallback = _ref2.holderCallback;
          var updateMark = index === notices.value.length - 1 ? notice.updateMark : undefined;
          var key = notice.key,
            userPassKey = notice.userPassKey;
          var content = notice.content;
          var noticeProps = _extends(_extends(_extends({
            prefixCls: prefixCls,
            closeIcon: typeof closeIcon === 'function' ? closeIcon({
              prefixCls: prefixCls
            }) : closeIcon
          }, notice), notice.props), {
            key: key,
            noticeKey: userPassKey || key,
            updateMark: updateMark,
            onClose: function onClose(noticeKey) {
              var _a;
              remove(noticeKey);
              (_a = notice.onClose) === null || _a === void 0 ? void 0 : _a.call(notice);
            },
            onClick: notice.onClick
          });
          if (holderCallback) {
            return _createVNode("div", {
              "key": key,
              "class": "".concat(prefixCls, "-hook-holder"),
              "ref": function ref(div) {
                if (typeof key === 'undefined') {
                  return;
                }
                if (div) {
                  hookRefs.set(key, div);
                  holderCallback(div, noticeProps);
                } else {
                  hookRefs.delete(key);
                }
              }
            }, null);
          }
          return _createVNode(Notice, _objectSpread(_objectSpread({}, noticeProps), {}, {
            "class": classNames(noticeProps.class, props.hashId)
          }), {
            default: function _default() {
              return [typeof content === 'function' ? content({
                prefixCls: prefixCls
              }) : content];
            }
          });
        });
        var className = (_className = {}, _defineProperty(_className, prefixCls, 1), _defineProperty(_className, "".concat(prefixCls, "-").concat(placement), 1), _defineProperty(_className, attrs.class, !!attrs.class), _defineProperty(_className, props.hashId, true), _defineProperty(_className, classes, !!classes), _className);
        function onAfterLeave() {
          var _a;
          if (noticesForPlacement.length > 0) {
            return;
          }
          Reflect.deleteProperty(placements.value, placement);
          (_a = props.onAllRemoved) === null || _a === void 0 ? void 0 : _a.call(props);
        }
        return _createVNode("div", {
          "key": placement,
          "class": className,
          "style": attrs.style || styles || {
            top: '65px',
            left: '50%'
          }
        }, [_createVNode(TransitionGroup, _objectSpread(_objectSpread({
          "tag": "div"
        }, transitionProps.value), {}, {
          "onAfterLeave": onAfterLeave
        }), {
          default: function _default() {
            return [noticeNodesForPlacement];
          }
        })]);
      });
      return _createVNode(Portal, {
        "getContainer": props.getContainer
      }, {
        default: function _default() {
          return [noticeNodes];
        }
      });
    };
  }
});
export default Notification;
