// ant-design-vue/lib/dropdown/style/button.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var genButtonStyle = function genButtonStyle(token) {
  var _$concat$concat;
  var componentCls = token.componentCls,
    antCls = token.antCls,
    paddingXS = token.paddingXS,
    opacityLoading = token.opacityLoading;
  return (0, _defineProperty2.default)({}, "".concat(componentCls, "-button"), (0, _defineProperty2.default)({
    whiteSpace: 'nowrap'
  }, "&".concat(antCls, "-btn-group > ").concat(antCls, "-btn"), (_$concat$concat = {}, (0, _defineProperty2.default)(_$concat$concat, "&-loading, &-loading + ".concat(antCls, "-btn"), {
    cursor: 'default',
    pointerEvents: 'none',
    opacity: opacityLoading
  }), (0, _defineProperty2.default)(_$concat$concat, "&:last-child:not(:first-child):not(".concat(antCls, "-btn-icon-only)"), {
    paddingInline: paddingXS
  }), _$concat$concat)));
};
var _default = genButtonStyle;
exports.default = _default;
