// ant-design-vue/lib/menu/style/horizontal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var getHorizontalStyle = function getHorizontalStyle(token) {
  var _$concat;
  var componentCls = token.componentCls,
    motionDurationSlow = token.motionDurationSlow,
    menuHorizontalHeight = token.menuHorizontalHeight,
    colorSplit = token.colorSplit,
    lineWidth = token.lineWidth,
    lineType = token.lineType,
    menuItemPaddingInline = token.menuItemPaddingInline;
  return (0, _defineProperty2.default)({}, "".concat(componentCls, "-horizontal"), (_$concat = {
    lineHeight: "".concat(menuHorizontalHeight, "px"),
    border: 0,
    borderBottom: "".concat(lineWidth, "px ").concat(lineType, " ").concat(colorSplit),
    boxShadow: 'none',
    '&::after': {
      display: 'block',
      clear: 'both',
      height: 0,
      content: '"\\20"'
    }
  }, (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-item, ").concat(componentCls, "-submenu"), {
    position: 'relative',
    display: 'inline-block',
    verticalAlign: 'bottom',
    paddingInline: menuItemPaddingInline
  }), (0, _defineProperty2.default)(_$concat, "> ".concat(componentCls, "-item:hover,\n        > ").concat(componentCls, "-item-active,\n        > ").concat(componentCls, "-submenu ").concat(componentCls, "-submenu-title:hover"), {
    backgroundColor: 'transparent'
  }), (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-item, ").concat(componentCls, "-submenu-title"), {
    transition: ["border-color ".concat(motionDurationSlow), "background ".concat(motionDurationSlow)].join(',')
  }), (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-submenu-arrow"), {
    display: 'none'
  }), _$concat));
};
var _default = getHorizontalStyle;
exports.default = _default;
