// ant-design-vue/lib/menu/style/theme.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _extends3 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _style = require("../../style");
var accessibilityFocus = function accessibilityFocus(token) {
  return (0, _extends3.default)({}, (0, _style.genFocusOutline)(token));
};
var getThemeStyle = function getThemeStyle(token, themeSuffix) {
  var _not$concat, _$concat3, _$concat4, _$concat$concat3, _$concat8, _$concat$concat4;
  var componentCls = token.componentCls,
    colorItemText = token.colorItemText,
    colorItemTextSelected = token.colorItemTextSelected,
    colorGroupTitle = token.colorGroupTitle,
    colorItemBg = token.colorItemBg,
    colorSubItemBg = token.colorSubItemBg,
    colorItemBgSelected = token.colorItemBgSelected,
    colorActiveBarHeight = token.colorActiveBarHeight,
    colorActiveBarWidth = token.colorActiveBarWidth,
    colorActiveBarBorderSize = token.colorActiveBarBorderSize,
    motionDurationSlow = token.motionDurationSlow,
    motionEaseInOut = token.motionEaseInOut,
    motionEaseOut = token.motionEaseOut,
    menuItemPaddingInline = token.menuItemPaddingInline,
    motionDurationMid = token.motionDurationMid,
    colorItemTextHover = token.colorItemTextHover,
    lineType = token.lineType,
    colorSplit = token.colorSplit,
    colorItemTextDisabled = token.colorItemTextDisabled,
    colorDangerItemText = token.colorDangerItemText,
    colorDangerItemTextHover = token.colorDangerItemTextHover,
    colorDangerItemTextSelected = token.colorDangerItemTextSelected,
    colorDangerItemBgActive = token.colorDangerItemBgActive,
    colorDangerItemBgSelected = token.colorDangerItemBgSelected,
    colorItemBgHover = token.colorItemBgHover,
    menuSubMenuBg = token.menuSubMenuBg,
    colorItemTextSelectedHorizontal = token.colorItemTextSelectedHorizontal,
    colorItemBgSelectedHorizontal = token.colorItemBgSelectedHorizontal;
  return (0, _defineProperty2.default)({}, "".concat(componentCls, "-").concat(themeSuffix), (_$concat$concat4 = {
    color: colorItemText,
    background: colorItemBg
  }, (0, _defineProperty2.default)(_$concat$concat4, "&".concat(componentCls, "-root:focus-visible"), (0, _extends3.default)({}, accessibilityFocus(token))), (0, _defineProperty2.default)(_$concat$concat4, "".concat(componentCls, "-item-group-title"), {
    color: colorGroupTitle
  }), (0, _defineProperty2.default)(_$concat$concat4, "".concat(componentCls, "-submenu-selected"), (0, _defineProperty2.default)({}, "> ".concat(componentCls, "-submenu-title"), {
    color: colorItemTextSelected
  })), (0, _defineProperty2.default)(_$concat$concat4, "".concat(componentCls, "-item-disabled, ").concat(componentCls, "-submenu-disabled"), {
    color: "".concat(colorItemTextDisabled, " !important")
  }), (0, _defineProperty2.default)(_$concat$concat4, "".concat(componentCls, "-item:hover, ").concat(componentCls, "-submenu-title:hover"), (0, _defineProperty2.default)({}, "&:not(".concat(componentCls, "-item-selected):not(").concat(componentCls, "-submenu-selected)"), {
    color: colorItemTextHover
  })), (0, _defineProperty2.default)(_$concat$concat4, "&:not(".concat(componentCls, "-horizontal)"), (_not$concat = {}, (0, _defineProperty2.default)(_not$concat, "".concat(componentCls, "-item:not(").concat(componentCls, "-item-selected)"), {
    '&:hover': {
      backgroundColor: colorItemBgHover
    },
    '&:active': {
      backgroundColor: colorItemBgSelected
    }
  }), (0, _defineProperty2.default)(_not$concat, "".concat(componentCls, "-submenu-title"), {
    '&:hover': {
      backgroundColor: colorItemBgHover
    },
    '&:active': {
      backgroundColor: colorItemBgSelected
    }
  }), _not$concat)), (0, _defineProperty2.default)(_$concat$concat4, "".concat(componentCls, "-item-danger"), (_$concat3 = {
    color: colorDangerItemText
  }, (0, _defineProperty2.default)(_$concat3, "&".concat(componentCls, "-item:hover"), (0, _defineProperty2.default)({}, "&:not(".concat(componentCls, "-item-selected):not(").concat(componentCls, "-submenu-selected)"), {
    color: colorDangerItemTextHover
  })), (0, _defineProperty2.default)(_$concat3, "&".concat(componentCls, "-item:active"), {
    background: colorDangerItemBgActive
  }), _$concat3)), (0, _defineProperty2.default)(_$concat$concat4, "".concat(componentCls, "-item a"), {
    '&, &:hover': {
      color: 'inherit'
    }
  }), (0, _defineProperty2.default)(_$concat$concat4, "".concat(componentCls, "-item-selected"), (_$concat4 = {
    color: colorItemTextSelected
  }, (0, _defineProperty2.default)(_$concat4, "&".concat(componentCls, "-item-danger"), {
    color: colorDangerItemTextSelected
  }), (0, _defineProperty2.default)(_$concat4, "a, a:hover", {
    color: 'inherit'
  }), _$concat4)), (0, _defineProperty2.default)(_$concat$concat4, "& ".concat(componentCls, "-item-selected"), (0, _defineProperty2.default)({
    backgroundColor: colorItemBgSelected
  }, "&".concat(componentCls, "-item-danger"), {
    backgroundColor: colorDangerItemBgSelected
  })), (0, _defineProperty2.default)(_$concat$concat4, "".concat(componentCls, "-item, ").concat(componentCls, "-submenu-title"), (0, _defineProperty2.default)({}, "&:not(".concat(componentCls, "-item-disabled):focus-visible"), (0, _extends3.default)({}, accessibilityFocus(token)))), (0, _defineProperty2.default)(_$concat$concat4, "&".concat(componentCls, "-submenu > ").concat(componentCls), {
    backgroundColor: menuSubMenuBg
  }), (0, _defineProperty2.default)(_$concat$concat4, "&".concat(componentCls, "-popup > ").concat(componentCls), {
    backgroundColor: colorItemBg
  }), (0, _defineProperty2.default)(_$concat$concat4, "&".concat(componentCls, "-horizontal"), (0, _extends3.default)((0, _extends3.default)({}, themeSuffix === 'dark' ? {
    borderBottom: 0
  } : {}), (0, _defineProperty2.default)({}, "> ".concat(componentCls, "-item, > ").concat(componentCls, "-submenu"), (_$concat$concat3 = {
    top: colorActiveBarBorderSize,
    marginTop: -colorActiveBarBorderSize,
    marginBottom: 0,
    borderRadius: 0,
    '&::after': {
      position: 'absolute',
      insetInline: menuItemPaddingInline,
      bottom: 0,
      borderBottom: "".concat(colorActiveBarHeight, "px solid transparent"),
      transition: "border-color ".concat(motionDurationSlow, " ").concat(motionEaseInOut),
      content: '""'
    }
  }, (0, _defineProperty2.default)(_$concat$concat3, "&:hover, &-active, &-open", {
    '&::after': {
      borderBottomWidth: colorActiveBarHeight,
      borderBottomColor: colorItemTextSelectedHorizontal
    }
  }), (0, _defineProperty2.default)(_$concat$concat3, "&-selected", {
    color: colorItemTextSelectedHorizontal,
    backgroundColor: colorItemBgSelectedHorizontal,
    '&::after': {
      borderBottomWidth: colorActiveBarHeight,
      borderBottomColor: colorItemTextSelectedHorizontal
    }
  }), _$concat$concat3)))), (0, _defineProperty2.default)(_$concat$concat4, "&".concat(componentCls, "-root"), (0, _defineProperty2.default)({}, "&".concat(componentCls, "-inline, &").concat(componentCls, "-vertical"), {
    borderInlineEnd: "".concat(colorActiveBarBorderSize, "px ").concat(lineType, " ").concat(colorSplit)
  })), (0, _defineProperty2.default)(_$concat$concat4, "&".concat(componentCls, "-inline"), (_$concat8 = {}, (0, _defineProperty2.default)(_$concat8, "".concat(componentCls, "-sub").concat(componentCls, "-inline"), {
    background: colorSubItemBg
  }), (0, _defineProperty2.default)(_$concat8, "".concat(componentCls, "-item, ").concat(componentCls, "-submenu-title"), colorActiveBarBorderSize && colorActiveBarWidth ? {
    width: "calc(100% + ".concat(colorActiveBarBorderSize, "px)")
  } : {}), (0, _defineProperty2.default)(_$concat8, "".concat(componentCls, "-item"), (0, _defineProperty2.default)({
    position: 'relative',
    '&::after': {
      position: 'absolute',
      insetBlock: 0,
      insetInlineEnd: 0,
      borderInlineEnd: "".concat(colorActiveBarWidth, "px solid ").concat(colorItemTextSelected),
      transform: 'scaleY(0.0001)',
      opacity: 0,
      transition: ["transform ".concat(motionDurationMid, " ").concat(motionEaseOut), "opacity ".concat(motionDurationMid, " ").concat(motionEaseOut)].join(','),
      content: '""'
    }
  }, "&".concat(componentCls, "-item-danger"), {
    '&::after': {
      borderInlineEndColor: colorDangerItemTextSelected
    }
  })), (0, _defineProperty2.default)(_$concat8, "".concat(componentCls, "-selected, ").concat(componentCls, "-item-selected"), {
    '&::after': {
      transform: 'scaleY(1)',
      opacity: 1,
      transition: ["transform ".concat(motionDurationMid, " ").concat(motionEaseInOut), "opacity ".concat(motionDurationMid, " ").concat(motionEaseInOut)].join(',')
    }
  }), _$concat8)), _$concat$concat4));
};
var _default = getThemeStyle;
exports.default = _default;
