// ant-design-vue/lib/popconfirm/style/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _internal = require("../../theme/internal");
// =============================== Base ===============================
var genBaseStyle = function genBaseStyle(token) {
  var _$concat, _componentCls;
  var componentCls = token.componentCls,
    iconCls = token.iconCls,
    zIndexPopup = token.zIndexPopup,
    colorText = token.colorText,
    colorWarning = token.colorWarning,
    marginXS = token.marginXS,
    fontSize = token.fontSize,
    fontWeightStrong = token.fontWeightStrong,
    lineHeight = token.lineHeight;
  return (0, _defineProperty2.default)({}, componentCls, (_componentCls = {
    zIndex: zIndexPopup
  }, (0, _defineProperty2.default)(_componentCls, "".concat(componentCls, "-inner-content"), {
    color: colorText
  }), (0, _defineProperty2.default)(_componentCls, "".concat(componentCls, "-message"), (_$concat = {
    position: 'relative',
    marginBottom: marginXS,
    color: colorText,
    fontSize: fontSize,
    display: 'flex',
    flexWrap: 'nowrap',
    alignItems: 'start'
  }, (0, _defineProperty2.default)(_$concat, "> ".concat(componentCls, "-message-icon ").concat(iconCls), {
    color: colorWarning,
    fontSize: fontSize,
    flex: 'none',
    lineHeight: 1,
    paddingTop: (Math.round(fontSize * lineHeight) - fontSize) / 2
  }), (0, _defineProperty2.default)(_$concat, '&-title', {
    flex: 'auto',
    marginInlineStart: marginXS
  }), (0, _defineProperty2.default)(_$concat, '&-title-only', {
    fontWeight: fontWeightStrong
  }), _$concat)), (0, _defineProperty2.default)(_componentCls, "".concat(componentCls, "-description"), {
    position: 'relative',
    marginInlineStart: fontSize + marginXS,
    marginBottom: marginXS,
    color: colorText,
    fontSize: fontSize
  }), (0, _defineProperty2.default)(_componentCls, "".concat(componentCls, "-buttons"), {
    textAlign: 'end',
    button: {
      marginInlineStart: marginXS
    }
  }), _componentCls));
};
// ============================== Export ==============================
var _default = (0, _internal.genComponentStyleHook)('Popconfirm', function (token) {
  return genBaseStyle(token);
}, function (token) {
  var zIndexPopupBase = token.zIndexPopupBase;
  return {
    zIndexPopup: zIndexPopupBase + 60
  };
});
exports.default = _default;
