// ant-design-vue/lib/table/style/bordered.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends4 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var genBorderedStyle = function genBorderedStyle(token) {
  var _table, _$concat2, _extends2, _$concat4, _$concat5;
  var componentCls = token.componentCls;
  var tableBorder = "".concat(token.lineWidth, "px ").concat(token.lineType, " ").concat(token.tableBorderColor);
  var getSizeBorderStyle = function getSizeBorderStyle(size, paddingVertical, paddingHorizontal) {
    return (0, _defineProperty2.default)({}, "&".concat(componentCls, "-").concat(size), (0, _defineProperty2.default)({}, "> ".concat(componentCls, "-container"), (0, _defineProperty2.default)({}, "> ".concat(componentCls, "-content, > ").concat(componentCls, "-body"), {
      '> table > tbody > tr > td': (0, _defineProperty2.default)({}, "> ".concat(componentCls, "-expanded-row-fixed"), {
        margin: "-".concat(paddingVertical, "px -").concat(paddingHorizontal + token.lineWidth, "px")
      })
    })));
  };
  return (0, _defineProperty2.default)({}, "".concat(componentCls, "-wrapper"), (_$concat5 = {}, (0, _defineProperty2.default)(_$concat5, "".concat(componentCls).concat(componentCls, "-bordered"), (0, _extends4.default)((0, _extends4.default)((0, _extends4.default)((_extends2 = {}, (0, _defineProperty2.default)(_extends2, "> ".concat(componentCls, "-title"), {
    border: tableBorder,
    borderBottom: 0
  }), (0, _defineProperty2.default)(_extends2, "> ".concat(componentCls, "-container"), (_$concat2 = {
    borderInlineStart: tableBorder
  }, (0, _defineProperty2.default)(_$concat2, "\n            > ".concat(componentCls, "-content,\n            > ").concat(componentCls, "-header,\n            > ").concat(componentCls, "-body,\n            > ").concat(componentCls, "-summary\n          "), {
    '> table': (_table = {}, (0, _defineProperty2.default)(_table, "\n                > thead > tr > th,\n                > tbody > tr > td,\n                > tfoot > tr > th,\n                > tfoot > tr > td\n              ", {
      borderInlineEnd: tableBorder
    }), (0, _defineProperty2.default)(_table, '> thead', {
      '> tr:not(:last-child) > th': {
        borderBottom: tableBorder
      },
      '> tr > th::before': {
        backgroundColor: 'transparent !important'
      }
    }), (0, _defineProperty2.default)(_table, "\n                > thead > tr,\n                > tbody > tr,\n                > tfoot > tr\n              ", (0, _defineProperty2.default)({}, "> ".concat(componentCls, "-cell-fix-right-first::after"), {
      borderInlineEnd: tableBorder
    })), (0, _defineProperty2.default)(_table, '> tbody > tr > td', (0, _defineProperty2.default)({}, "> ".concat(componentCls, "-expanded-row-fixed"), {
      margin: "-".concat(token.tablePaddingVertical, "px -").concat(token.tablePaddingHorizontal + token.lineWidth, "px"),
      '&::after': {
        position: 'absolute',
        top: 0,
        insetInlineEnd: token.lineWidth,
        bottom: 0,
        borderInlineEnd: tableBorder,
        content: '""'
      }
    })), _table)
  }), (0, _defineProperty2.default)(_$concat2, "\n            > ".concat(componentCls, "-content,\n            > ").concat(componentCls, "-header\n          "), {
    '> table': {
      borderTop: tableBorder
    }
  }), _$concat2)), (0, _defineProperty2.default)(_extends2, "&".concat(componentCls, "-scroll-horizontal"), (0, _defineProperty2.default)({}, "> ".concat(componentCls, "-container > ").concat(componentCls, "-body"), {
    '> table > tbody': (0, _defineProperty2.default)({}, "\n                > tr".concat(componentCls, "-expanded-row,\n                > tr").concat(componentCls, "-placeholder\n              "), {
      '> td': {
        borderInlineEnd: 0
      }
    })
  })), _extends2), getSizeBorderStyle('middle', token.tablePaddingVerticalMiddle, token.tablePaddingHorizontalMiddle)), getSizeBorderStyle('small', token.tablePaddingVerticalSmall, token.tablePaddingHorizontalSmall)), (0, _defineProperty2.default)({}, "> ".concat(componentCls, "-footer"), {
    border: tableBorder,
    borderTop: 0
  }))), (0, _defineProperty2.default)(_$concat5, "".concat(componentCls, "-cell"), (_$concat4 = {}, (0, _defineProperty2.default)(_$concat4, "".concat(componentCls, "-container:first-child"), {
    // :first-child to avoid the case when bordered and title is set
    borderTop: 0
  }), (0, _defineProperty2.default)(_$concat4, '&-scrollbar:not([rowspan])', {
    boxShadow: "0 ".concat(token.lineWidth, "px 0 ").concat(token.lineWidth, "px ").concat(token.tableHeaderBg)
  }), _$concat4)), _$concat5));
};
var _default = genBorderedStyle;
exports.default = _default;
