// ant-design-vue/lib/table/style/filter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends3 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _style = require("../../style");
var genFilterStyle = function genFilterStyle(token) {
  var _$concat, _$concat2, _extends2, _$concat$concat$conca;
  var componentCls = token.componentCls,
    antCls = token.antCls,
    iconCls = token.iconCls,
    tableFilterDropdownWidth = token.tableFilterDropdownWidth,
    tableFilterDropdownSearchWidth = token.tableFilterDropdownSearchWidth,
    paddingXXS = token.paddingXXS,
    paddingXS = token.paddingXS,
    colorText = token.colorText,
    lineWidth = token.lineWidth,
    lineType = token.lineType,
    tableBorderColor = token.tableBorderColor,
    tableHeaderIconColor = token.tableHeaderIconColor,
    fontSizeSM = token.fontSizeSM,
    tablePaddingHorizontal = token.tablePaddingHorizontal,
    borderRadius = token.borderRadius,
    motionDurationSlow = token.motionDurationSlow,
    colorTextDescription = token.colorTextDescription,
    colorPrimary = token.colorPrimary,
    tableHeaderFilterActiveBg = token.tableHeaderFilterActiveBg,
    colorTextDisabled = token.colorTextDisabled,
    tableFilterDropdownBg = token.tableFilterDropdownBg,
    tableFilterDropdownHeight = token.tableFilterDropdownHeight,
    controlItemBgHover = token.controlItemBgHover,
    controlItemBgActive = token.controlItemBgActive,
    boxShadowSecondary = token.boxShadowSecondary;
  var dropdownPrefixCls = "".concat(antCls, "-dropdown");
  var tableFilterDropdownPrefixCls = "".concat(componentCls, "-filter-dropdown");
  var treePrefixCls = "".concat(antCls, "-tree");
  var tableBorder = "".concat(lineWidth, "px ").concat(lineType, " ").concat(tableBorderColor);
  return [(0, _defineProperty2.default)({}, "".concat(componentCls, "-wrapper"), (_$concat = {}, (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-filter-column"), {
    display: 'flex',
    justifyContent: 'space-between'
  }), (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-filter-trigger"), {
    position: 'relative',
    display: 'flex',
    alignItems: 'center',
    marginBlock: -paddingXXS,
    marginInline: "".concat(paddingXXS, "px ").concat(-tablePaddingHorizontal / 2, "px"),
    padding: "0 ".concat(paddingXXS, "px"),
    color: tableHeaderIconColor,
    fontSize: fontSizeSM,
    borderRadius: borderRadius,
    cursor: 'pointer',
    transition: "all ".concat(motionDurationSlow),
    '&:hover': {
      color: colorTextDescription,
      background: tableHeaderFilterActiveBg
    },
    '&.active': {
      color: colorPrimary
    }
  }), _$concat)), (0, _defineProperty2.default)({}, "".concat(antCls, "-dropdown"), (0, _defineProperty2.default)({}, tableFilterDropdownPrefixCls, (0, _extends3.default)((0, _extends3.default)({}, (0, _style.resetComponent)(token)), (_extends2 = {
    minWidth: tableFilterDropdownWidth,
    backgroundColor: tableFilterDropdownBg,
    borderRadius: borderRadius,
    boxShadow: boxShadowSecondary
  }, (0, _defineProperty2.default)(_extends2, "".concat(dropdownPrefixCls, "-menu"), {
    // https://github.com/ant-design/ant-design/issues/4916
    // https://github.com/ant-design/ant-design/issues/19542
    maxHeight: tableFilterDropdownHeight,
    overflowX: 'hidden',
    border: 0,
    boxShadow: 'none',
    '&:empty::after': {
      display: 'block',
      padding: "".concat(paddingXS, "px 0"),
      color: colorTextDisabled,
      fontSize: fontSizeSM,
      textAlign: 'center',
      content: '"Not Found"'
    }
  }), (0, _defineProperty2.default)(_extends2, "".concat(tableFilterDropdownPrefixCls, "-tree"), (_$concat2 = {
    paddingBlock: "".concat(paddingXS, "px 0"),
    paddingInline: paddingXS
  }, (0, _defineProperty2.default)(_$concat2, treePrefixCls, {
    padding: 0
  }), (0, _defineProperty2.default)(_$concat2, "".concat(treePrefixCls, "-treenode ").concat(treePrefixCls, "-node-content-wrapper:hover"), {
    backgroundColor: controlItemBgHover
  }), (0, _defineProperty2.default)(_$concat2, "".concat(treePrefixCls, "-treenode-checkbox-checked ").concat(treePrefixCls, "-node-content-wrapper"), {
    '&, &:hover': {
      backgroundColor: controlItemBgActive
    }
  }), _$concat2)), (0, _defineProperty2.default)(_extends2, "".concat(tableFilterDropdownPrefixCls, "-search"), {
    padding: paddingXS,
    borderBottom: tableBorder,
    '&-input': (0, _defineProperty2.default)({
      input: {
        minWidth: tableFilterDropdownSearchWidth
      }
    }, iconCls, {
      color: colorTextDisabled
    })
  }), (0, _defineProperty2.default)(_extends2, "".concat(tableFilterDropdownPrefixCls, "-checkall"), {
    width: '100%',
    marginBottom: paddingXXS,
    marginInlineStart: paddingXXS
  }), (0, _defineProperty2.default)(_extends2, "".concat(tableFilterDropdownPrefixCls, "-btns"), {
    display: 'flex',
    justifyContent: 'space-between',
    padding: "".concat(paddingXS - lineWidth, "px ").concat(paddingXS, "px"),
    overflow: 'hidden',
    backgroundColor: 'inherit',
    borderTop: tableBorder
  }), _extends2)))), // Dropdown Menu & SubMenu
  (0, _defineProperty2.default)({}, "".concat(antCls, "-dropdown ").concat(tableFilterDropdownPrefixCls, ", ").concat(tableFilterDropdownPrefixCls, "-submenu"), (_$concat$concat$conca = {}, (0, _defineProperty2.default)(_$concat$concat$conca, "".concat(antCls, "-checkbox-wrapper + span"), {
    paddingInlineStart: paddingXS,
    color: colorText
  }), (0, _defineProperty2.default)(_$concat$concat$conca, "> ul", {
    maxHeight: 'calc(100vh - 130px)',
    overflowX: 'hidden',
    overflowY: 'auto'
  }), _$concat$concat$conca))];
};
var _default = genFilterStyle;
exports.default = _default;
