// ant-design-vue/lib/table/style/fixed.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var genFixedStyle = function genFixedStyle(token) {
  var _$concat, _$concat2, _$concat3;
  var componentCls = token.componentCls,
    lineWidth = token.lineWidth,
    colorSplit = token.colorSplit,
    motionDurationSlow = token.motionDurationSlow,
    zIndexTableFixed = token.zIndexTableFixed,
    tableBg = token.tableBg,
    zIndexTableSticky = token.zIndexTableSticky;
  var shadowColor = colorSplit;
  // Follow style is magic of shadow which should not follow token:
  return (0, _defineProperty2.default)({}, "".concat(componentCls, "-wrapper"), (_$concat3 = {}, (0, _defineProperty2.default)(_$concat3, "\n        ".concat(componentCls, "-cell-fix-left,\n        ").concat(componentCls, "-cell-fix-right\n      "), {
    position: 'sticky !important',
    zIndex: zIndexTableFixed,
    background: tableBg
  }), (0, _defineProperty2.default)(_$concat3, "\n        ".concat(componentCls, "-cell-fix-left-first::after,\n        ").concat(componentCls, "-cell-fix-left-last::after\n      "), {
    position: 'absolute',
    top: 0,
    right: {
      _skip_check_: true,
      value: 0
    },
    bottom: -lineWidth,
    width: 30,
    transform: 'translateX(100%)',
    transition: "box-shadow ".concat(motionDurationSlow),
    content: '""',
    pointerEvents: 'none'
  }), (0, _defineProperty2.default)(_$concat3, "".concat(componentCls, "-cell-fix-left-all::after"), {
    display: 'none'
  }), (0, _defineProperty2.default)(_$concat3, "\n        ".concat(componentCls, "-cell-fix-right-first::after,\n        ").concat(componentCls, "-cell-fix-right-last::after\n      "), {
    position: 'absolute',
    top: 0,
    bottom: -lineWidth,
    left: {
      _skip_check_: true,
      value: 0
    },
    width: 30,
    transform: 'translateX(-100%)',
    transition: "box-shadow ".concat(motionDurationSlow),
    content: '""',
    pointerEvents: 'none'
  }), (0, _defineProperty2.default)(_$concat3, "".concat(componentCls, "-container"), {
    '&::before, &::after': {
      position: 'absolute',
      top: 0,
      bottom: 0,
      zIndex: zIndexTableSticky + 1,
      width: 30,
      transition: "box-shadow ".concat(motionDurationSlow),
      content: '""',
      pointerEvents: 'none'
    },
    '&::before': {
      insetInlineStart: 0
    },
    '&::after': {
      insetInlineEnd: 0
    }
  }), (0, _defineProperty2.default)(_$concat3, "".concat(componentCls, "-ping-left"), (_$concat = {}, (0, _defineProperty2.default)(_$concat, "&:not(".concat(componentCls, "-has-fix-left) ").concat(componentCls, "-container"), {
    position: 'relative',
    '&::before': {
      boxShadow: "inset 10px 0 8px -8px ".concat(shadowColor)
    }
  }), (0, _defineProperty2.default)(_$concat, "\n          ".concat(componentCls, "-cell-fix-left-first::after,\n          ").concat(componentCls, "-cell-fix-left-last::after\n        "), {
    boxShadow: "inset 10px 0 8px -8px ".concat(shadowColor)
  }), (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-cell-fix-left-last::before"), {
    backgroundColor: 'transparent !important'
  }), _$concat)), (0, _defineProperty2.default)(_$concat3, "".concat(componentCls, "-ping-right"), (_$concat2 = {}, (0, _defineProperty2.default)(_$concat2, "&:not(".concat(componentCls, "-has-fix-right) ").concat(componentCls, "-container"), {
    position: 'relative',
    '&::after': {
      boxShadow: "inset -10px 0 8px -8px ".concat(shadowColor)
    }
  }), (0, _defineProperty2.default)(_$concat2, "\n          ".concat(componentCls, "-cell-fix-right-first::after,\n          ").concat(componentCls, "-cell-fix-right-last::after\n        "), {
    boxShadow: "inset -10px 0 8px -8px ".concat(shadowColor)
  }), _$concat2)), _$concat3));
};
var _default = genFixedStyle;
exports.default = _default;
