// ant-design-vue/lib/table/style/pagination.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var genPaginationStyle = function genPaginationStyle(token) {
  var _$concat;
  var componentCls = token.componentCls,
    antCls = token.antCls;
  return (0, _defineProperty2.default)({}, "".concat(componentCls, "-wrapper"), (_$concat = {}, (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-pagination").concat(antCls, "-pagination"), {
    margin: "".concat(token.margin, "px 0")
  }), (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-pagination"), {
    display: 'flex',
    flexWrap: 'wrap',
    rowGap: token.paddingXS,
    '> *': {
      flex: 'none'
    },
    '&-left': {
      justifyContent: 'flex-start'
    },
    '&-center': {
      justifyContent: 'center'
    },
    '&-right': {
      justifyContent: 'flex-end'
    }
  }), _$concat));
};
var _default = genPaginationStyle;
exports.default = _default;
