// ant-design-vue/lib/table/style/resize.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var genResizeStyle = function genResizeStyle(token) {
  var _$concat$concat, _$concat$concat2, _ref;
  var componentCls = token.componentCls;
  return _ref = {}, (0, _defineProperty2.default)(_ref, "".concat(componentCls, "-wrapper ").concat(componentCls, "-resize-handle"), (_$concat$concat = {
    position: 'absolute',
    top: 0,
    height: '100% !important',
    bottom: 0,
    left: ' auto !important',
    right: ' -8px',
    cursor: 'col-resize',
    touchAction: 'none',
    userSelect: 'auto',
    width: '16px',
    zIndex: 1
  }, (0, _defineProperty2.default)(_$concat$concat, "&-line", {
    display: 'block',
    width: '1px',
    marginLeft: '7px',
    height: '100% !important',
    backgroundColor: token.colorPrimary,
    opacity: 0
  }), (0, _defineProperty2.default)(_$concat$concat, "&:hover &-line", {
    opacity: 1
  }), _$concat$concat)), (0, _defineProperty2.default)(_ref, "".concat(componentCls, "-wrapper  ").concat(componentCls, "-resize-handle.dragging"), (_$concat$concat2 = {
    overflow: 'hidden'
  }, (0, _defineProperty2.default)(_$concat$concat2, "".concat(componentCls, "-resize-handle-line"), {
    opacity: 1
  }), (0, _defineProperty2.default)(_$concat$concat2, "&:before", {
    position: 'absolute',
    top: 0,
    bottom: 0,
    content: '" "',
    width: '200vw',
    transform: 'translateX(-50%)',
    opacity: 0
  }), _$concat$concat2)), _ref;
};
var _default = genResizeStyle;
exports.default = _default;
