// ant-design-vue/lib/table/style/selection.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var genSelectionStyle = function genSelectionStyle(token) {
  var _$concat2;
  var componentCls = token.componentCls,
    antCls = token.antCls,
    iconCls = token.iconCls,
    fontSizeIcon = token.fontSizeIcon,
    paddingXS = token.paddingXS,
    tableHeaderIconColor = token.tableHeaderIconColor,
    tableHeaderIconColorHover = token.tableHeaderIconColorHover;
  return (0, _defineProperty2.default)({}, "".concat(componentCls, "-wrapper"), (_$concat2 = {}, (0, _defineProperty2.default)(_$concat2, "".concat(componentCls, "-selection-col"), {
    width: token.tableSelectionColumnWidth
  }), (0, _defineProperty2.default)(_$concat2, "".concat(componentCls, "-bordered ").concat(componentCls, "-selection-col"), {
    width: token.tableSelectionColumnWidth + paddingXS * 2
  }), (0, _defineProperty2.default)(_$concat2, "\n        table tr th".concat(componentCls, "-selection-column,\n        table tr td").concat(componentCls, "-selection-column\n      "), (0, _defineProperty2.default)({
    paddingInlineEnd: token.paddingXS,
    paddingInlineStart: token.paddingXS,
    textAlign: 'center'
  }, "".concat(antCls, "-radio-wrapper"), {
    marginInlineEnd: 0
  })), (0, _defineProperty2.default)(_$concat2, "table tr th".concat(componentCls, "-selection-column").concat(componentCls, "-cell-fix-left"), {
    zIndex: token.zIndexTableFixed + 1
  }), (0, _defineProperty2.default)(_$concat2, "table tr th".concat(componentCls, "-selection-column::after"), {
    backgroundColor: 'transparent !important'
  }), (0, _defineProperty2.default)(_$concat2, "".concat(componentCls, "-selection"), {
    position: 'relative',
    display: 'inline-flex',
    flexDirection: 'column'
  }), (0, _defineProperty2.default)(_$concat2, "".concat(componentCls, "-selection-extra"), (0, _defineProperty2.default)({
    position: 'absolute',
    top: 0,
    zIndex: 1,
    cursor: 'pointer',
    transition: "all ".concat(token.motionDurationSlow),
    marginInlineStart: '100%',
    paddingInlineStart: "".concat(token.tablePaddingHorizontal / 4, "px")
  }, iconCls, {
    color: tableHeaderIconColor,
    fontSize: fontSizeIcon,
    verticalAlign: 'baseline',
    '&:hover': {
      color: tableHeaderIconColorHover
    }
  })), _$concat2));
};
var _default = genSelectionStyle;
exports.default = _default;
