// ant-design-vue/lib/table/style/size.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var genSizeStyle = function genSizeStyle(token) {
  var componentCls = token.componentCls;
  var getSizeStyle = function getSizeStyle(size, paddingVertical, paddingHorizontal, fontSize) {
    var _$concat$concat$conca;
    return (0, _defineProperty2.default)({}, "".concat(componentCls).concat(componentCls, "-").concat(size), (_$concat$concat$conca = {
      fontSize: fontSize
    }, (0, _defineProperty2.default)(_$concat$concat$conca, "\n        ".concat(componentCls, "-title,\n        ").concat(componentCls, "-footer,\n        ").concat(componentCls, "-thead > tr > th,\n        ").concat(componentCls, "-tbody > tr > td,\n        tfoot > tr > th,\n        tfoot > tr > td\n      "), {
      padding: "".concat(paddingVertical, "px ").concat(paddingHorizontal, "px")
    }), (0, _defineProperty2.default)(_$concat$concat$conca, "".concat(componentCls, "-filter-trigger"), {
      marginInlineEnd: "-".concat(paddingHorizontal / 2, "px")
    }), (0, _defineProperty2.default)(_$concat$concat$conca, "".concat(componentCls, "-expanded-row-fixed"), {
      margin: "-".concat(paddingVertical, "px -").concat(paddingHorizontal, "px")
    }), (0, _defineProperty2.default)(_$concat$concat$conca, "".concat(componentCls, "-tbody"), (0, _defineProperty2.default)({}, "".concat(componentCls, "-wrapper:only-child ").concat(componentCls), {
      marginBlock: "-".concat(paddingVertical, "px"),
      marginInline: "".concat(token.tableExpandColumnWidth - paddingHorizontal, "px -").concat(paddingHorizontal, "px")
    })), (0, _defineProperty2.default)(_$concat$concat$conca, "".concat(componentCls, "-selection-column"), {
      paddingInlineStart: "".concat(paddingHorizontal / 4, "px")
    }), _$concat$concat$conca));
  };
  return (0, _defineProperty2.default)({}, "".concat(componentCls, "-wrapper"), (0, _extends2.default)((0, _extends2.default)({}, getSizeStyle('middle', token.tablePaddingVerticalMiddle, token.tablePaddingHorizontalMiddle, token.tableFontSizeMiddle)), getSizeStyle('small', token.tablePaddingVerticalSmall, token.tablePaddingHorizontalSmall, token.tableFontSizeSmall)));
};
var _default = genSizeStyle;
exports.default = _default;
