// ant-design-vue/lib/table/style/summary.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var genSummaryStyle = function genSummaryStyle(token) {
  var _$concat;
  var componentCls = token.componentCls,
    lineWidth = token.lineWidth,
    tableBorderColor = token.tableBorderColor;
  var tableBorder = "".concat(lineWidth, "px ").concat(token.lineType, " ").concat(tableBorderColor);
  return (0, _defineProperty2.default)({}, "".concat(componentCls, "-wrapper"), (_$concat = {}, (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-summary"), {
    position: 'relative',
    zIndex: token.zIndexTableFixed,
    background: token.tableBg,
    '> tr': {
      '> th, > td': {
        borderBottom: tableBorder
      }
    }
  }), (0, _defineProperty2.default)(_$concat, "div".concat(componentCls, "-summary"), {
    boxShadow: "0 -".concat(lineWidth, "px 0 ").concat(tableBorderColor)
  }), _$concat));
};
var _default = genSummaryStyle;
exports.default = _default;
