// ant-design-vue/es/_util/cssinjs/index.js
import useCacheToken from './hooks/useCacheToken';
import useStyleRegister, { extractStyle } from './hooks/useStyleRegister';
import Keyframes from './Keyframes';
import { legacyNotSelectorLinter, logicalPropertiesLinter } from './linters';
import { createCache, useStyleInject, useStyleProvider, StyleProvider } from './StyleContext';
import { createTheme, Theme } from './theme';
import legacyLogicalPropertiesTransformer from './transformers/legacyLogicalProperties';
var cssinjs = {
  Theme: Theme,
  createTheme: createTheme,
  useStyleRegister: useStyleRegister,
  useCacheToken: useCacheToken,
  createCache: createCache,
  useStyleInject: useStyleInject,
  useStyleProvider: useStyleProvider,
  Keyframes: Keyframes,
  extractStyle: extractStyle,
  // Transformer
  legacyLogicalPropertiesTransformer: legacyLogicalPropertiesTransformer,
  // Linters
  logicalPropertiesLinter: logicalPropertiesLinter,
  legacyNotSelectorLinter: legacyNotSelectorLinter,
  // cssinjs
  StyleProvider: StyleProvider
};
export { Theme, createTheme, useStyleRegister, useCacheToken, createCache, useStyleInject, useStyleProvider, Keyframes, extractStyle,
// Transformer
legacyLogicalPropertiesTransformer,
// Linters
logicalPropertiesLinter, legacyNotSelectorLinter,
// cssinjs
StyleProvider };
export default cssinjs;
