// ant-design-vue/lib/message/PurePanel.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.TypeIcon = exports.PureContent = void 0;
var _vue = require("vue");
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _Notice = _interopRequireDefault(require("../vc-notification/Notice"));
var _style = _interopRequireDefault(require("./style"));
var _LoadingOutlined = _interopRequireDefault(require("@ant-design/icons-vue/lib/icons/LoadingOutlined"));
var _ExclamationCircleFilled = _interopRequireDefault(require("@ant-design/icons-vue/lib/icons/ExclamationCircleFilled"));
var _CloseCircleFilled = _interopRequireDefault(require("@ant-design/icons-vue/lib/icons/CloseCircleFilled"));
var _CheckCircleFilled = _interopRequireDefault(require("@ant-design/icons-vue/lib/icons/CheckCircleFilled"));
var _InfoCircleFilled = _interopRequireDefault(require("@ant-design/icons-vue/lib/icons/InfoCircleFilled"));
var _classNames = _interopRequireDefault(require("../_util/classNames"));
var _context = require("../config-provider/context");
var TypeIcon = {
  info: (0, _vue.createVNode)(_InfoCircleFilled.default, null, null),
  success: (0, _vue.createVNode)(_CheckCircleFilled.default, null, null),
  error: (0, _vue.createVNode)(_CloseCircleFilled.default, null, null),
  warning: (0, _vue.createVNode)(_ExclamationCircleFilled.default, null, null),
  loading: (0, _vue.createVNode)(_LoadingOutlined.default, null, null)
};
exports.TypeIcon = TypeIcon;
var PureContent = (0, _vue.defineComponent)({
  name: 'PureContent',
  inheritAttrs: false,
  props: ['prefixCls', 'type', 'icon'],
  setup: function setup(props, _ref) {
    var slots = _ref.slots;
    return function () {
      var _a;
      return (0, _vue.createVNode)("div", {
        "class": (0, _classNames.default)("".concat(props.prefixCls, "-custom-content"), "".concat(props.prefixCls, "-").concat(props.type))
      }, [props.icon || TypeIcon[props.type], (0, _vue.createVNode)("span", null, [(_a = slots.default) === null || _a === void 0 ? void 0 : _a.call(slots)])]);
    };
  }
});
/** @private Internal Component. Do not use in your production. */
exports.PureContent = PureContent;
var _default2 = (0, _vue.defineComponent)({
  name: 'PurePanel',
  inheritAttrs: false,
  props: ['prefixCls', 'class', 'type', 'icon', 'content'],
  setup: function setup(props, _ref2) {
    var slots = _ref2.slots,
      attrs = _ref2.attrs;
    var _a;
    var _useConfigContextInje = (0, _context.useConfigContextInject)(),
      getPrefixCls = _useConfigContextInje.getPrefixCls;
    var prefixCls = (0, _vue.computed)(function () {
      return props.prefixCls || getPrefixCls('message');
    });
    var _useStyle = (0, _style.default)(prefixCls),
      _useStyle2 = (0, _slicedToArray2.default)(_useStyle, 2),
      hashId = _useStyle2[1];
    return (0, _vue.createVNode)(_Notice.default, (0, _objectSpread2.default)((0, _objectSpread2.default)({}, attrs), {}, {
      "prefixCls": prefixCls.value,
      "class": (0, _classNames.default)(hashId.value, "".concat(prefixCls.value, "-notice-pure-panel")),
      "noticeKey": "pure",
      "duration": null
    }), {
      default: function _default() {
        return [(0, _vue.createVNode)(PureContent, {
          "prefixCls": prefixCls.value,
          "type": props.type,
          "icon": props.icon
        }, {
          default: function _default() {
            return [(_a = slots.default) === null || _a === void 0 ? void 0 : _a.call(slots)];
          }
        })];
      }
    });
  }
});
exports.default = _default2;
