// ant-design-vue/es/message/PurePanel.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import { createVNode as _createVNode } from "vue";
import Notice from '../vc-notification/Notice';
import useStyle from './style';
import LoadingOutlined from "@ant-design/icons-vue/es/icons/LoadingOutlined";
import ExclamationCircleFilled from "@ant-design/icons-vue/es/icons/ExclamationCircleFilled";
import CloseCircleFilled from "@ant-design/icons-vue/es/icons/CloseCircleFilled";
import CheckCircleFilled from "@ant-design/icons-vue/es/icons/CheckCircleFilled";
import InfoCircleFilled from "@ant-design/icons-vue/es/icons/InfoCircleFilled";
import classNames from '../_util/classNames';
import { useConfigContextInject } from '../config-provider/context';
import { computed, defineComponent } from 'vue';
export var TypeIcon = {
  info: _createVNode(InfoCircleFilled, null, null),
  success: _createVNode(CheckCircleFilled, null, null),
  error: _createVNode(CloseCircleFilled, null, null),
  warning: _createVNode(ExclamationCircleFilled, null, null),
  loading: _createVNode(LoadingOutlined, null, null)
};
export var PureContent = defineComponent({
  name: 'PureContent',
  inheritAttrs: false,
  props: ['prefixCls', 'type', 'icon'],
  setup: function setup(props, _ref) {
    var slots = _ref.slots;
    return function () {
      var _a;
      return _createVNode("div", {
        "class": classNames("".concat(props.prefixCls, "-custom-content"), "".concat(props.prefixCls, "-").concat(props.type))
      }, [props.icon || TypeIcon[props.type], _createVNode("span", null, [(_a = slots.default) === null || _a === void 0 ? void 0 : _a.call(slots)])]);
    };
  }
});
/** @private Internal Component. Do not use in your production. */
export default defineComponent({
  name: 'PurePanel',
  inheritAttrs: false,
  props: ['prefixCls', 'class', 'type', 'icon', 'content'],
  setup: function setup(props, _ref2) {
    var slots = _ref2.slots,
      attrs = _ref2.attrs;
    var _a;
    var _useConfigContextInje = useConfigContextInject(),
      getPrefixCls = _useConfigContextInje.getPrefixCls;
    var prefixCls = computed(function () {
      return props.prefixCls || getPrefixCls('message');
    });
    var _useStyle = useStyle(prefixCls),
      _useStyle2 = _slicedToArray(_useStyle, 2),
      hashId = _useStyle2[1];
    return _createVNode(Notice, _objectSpread(_objectSpread({}, attrs), {}, {
      "prefixCls": prefixCls.value,
      "class": classNames(hashId.value, "".concat(prefixCls.value, "-notice-pure-panel")),
      "noticeKey": "pure",
      "duration": null
    }), {
      default: function _default() {
        return [_createVNode(PureContent, {
          "prefixCls": prefixCls.value,
          "type": props.type,
          "icon": props.icon
        }, {
          default: function _default() {
            return [(_a = slots.default) === null || _a === void 0 ? void 0 : _a.call(slots)];
          }
        })];
      }
    });
  }
});
