// ant-design-vue/es/message/useMessage.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import _typeof from "@babel/runtime/helpers/esm/typeof";
import _extends from "@babel/runtime/helpers/esm/extends";
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import { createVNode as _createVNode } from "vue";
var __rest = this && this.__rest || function (s, e) {
  var t = {};
  for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};
import { shallowRef, computed, defineComponent } from 'vue';
import { useNotification as useVcNotification } from '../vc-notification';
import CloseOutlined from '@ant-design/icons-vue';
import useStyle from './style';
import { PureContent } from './PurePanel';
import { getMotion } from '../vc-trigger/utils/motionUtil';
import { wrapPromiseFn } from '../_util/util';
import useConfigInject from '../config-provider/hooks/useConfigInject';
import classNames from '../_util/classNames';
var DEFAULT_OFFSET = 8;
var DEFAULT_DURATION = 3;
var Holder = defineComponent({
  name: 'Holder',
  inheritAttrs: false,
  props: ['top', 'prefixCls', 'getContainer', 'maxCount', 'duration', 'rtl', 'transitionName', 'onAllRemoved'],
  setup: function setup(props, _ref) {
    var expose = _ref.expose;
    var _a;
    var _useConfigInject = useConfigInject('message', props),
      getPrefixCls = _useConfigInject.getPrefixCls,
      getPopupContainer = _useConfigInject.getPopupContainer;
    var prefixCls = computed(function () {
      return getPrefixCls('message', props.prefixCls);
    });
    var _useStyle = useStyle(prefixCls),
      _useStyle2 = _slicedToArray(_useStyle, 2),
      hashId = _useStyle2[1];
    // =============================== Style ===============================
    var getStyles = function getStyles() {
      var _a;
      var top = (_a = props.top) !== null && _a !== void 0 ? _a : DEFAULT_OFFSET;
      return {
        left: '50%',
        transform: 'translateX(-50%)',
        top: typeof top === 'number' ? "".concat(top, "px") : top
      };
    };
    var getClassName = function getClassName() {
      return classNames(hashId.value, props.rtl ? "".concat(prefixCls.value, "-rtl") : '');
    };
    // ============================== Motion ===============================
    var getNotificationMotion = function getNotificationMotion() {
      var _a;
      return getMotion({
        prefixCls: prefixCls.value,
        animation: (_a = props.animation) !== null && _a !== void 0 ? _a : "move-up",
        transitionName: props.transitionName
      });
    };
    // ============================ Close Icon =============================
    var mergedCloseIcon = _createVNode("span", {
      "class": "".concat(prefixCls.value, "-close-x")
    }, [_createVNode(CloseOutlined, {
      "class": "".concat(prefixCls.value, "-close-icon")
    }, null)]);
    // ============================== Origin ===============================
    var _useVcNotification = useVcNotification({
        //@ts-ignore
        getStyles: getStyles,
        prefixCls: prefixCls.value,
        getClassName: getClassName,
        motion: getNotificationMotion,
        closable: false,
        closeIcon: mergedCloseIcon,
        duration: (_a = props.duration) !== null && _a !== void 0 ? _a : DEFAULT_DURATION,
        getContainer: function getContainer() {
          var _a, _b;
          return ((_a = props.staticGetContainer) === null || _a === void 0 ? void 0 : _a.call(props)) || ((_b = getPopupContainer.value) === null || _b === void 0 ? void 0 : _b.call(getPopupContainer)) || document.body;
        },
        maxCount: props.maxCount,
        onAllRemoved: props.onAllRemoved
      }),
      _useVcNotification2 = _slicedToArray(_useVcNotification, 2),
      api = _useVcNotification2[0],
      holder = _useVcNotification2[1];
    // ================================ Ref ================================
    expose(_extends(_extends({}, api), {
      prefixCls: prefixCls,
      hashId: hashId
    }));
    return holder;
  }
});
// ==============================================================================
// ==                                   Hook                                   ==
// ==============================================================================
var keyIndex = 0;
export function useInternalMessage(messageConfig) {
  var holderRef = shallowRef(null);
  var holderKey = Symbol('messageHolderKey');
  // ================================ API ================================
  // Wrap with notification content
  // >>> close
  var close = function close(key) {
    var _a;
    (_a = holderRef.value) === null || _a === void 0 ? void 0 : _a.close(key);
  };
  // >>> Open
  var open = function open(config) {
    if (!holderRef.value) {
      var fakeResult = function fakeResult() {};
      fakeResult.then = function () {};
      return fakeResult;
    }
    var _holderRef$value = holderRef.value,
      originOpen = _holderRef$value.open,
      prefixCls = _holderRef$value.prefixCls,
      hashId = _holderRef$value.hashId;
    var noticePrefixCls = "".concat(prefixCls, "-notice");
    var _content = config.content,
      icon = config.icon,
      type = config.type,
      key = config.key,
      className = config.class,
      _onClose = config.onClose,
      restConfig = __rest(config, ["content", "icon", "type", "key", "class", "onClose"]);
    var mergedKey = key;
    if (mergedKey === undefined || mergedKey === null) {
      keyIndex += 1;
      mergedKey = "antd-message-".concat(keyIndex);
    }
    return wrapPromiseFn(function (resolve) {
      originOpen(_extends(_extends({}, restConfig), {
        key: mergedKey,
        content: function content() {
          return _createVNode(PureContent, {
            "prefixCls": prefixCls,
            "type": type,
            "icon": typeof icon === 'function' ? icon() : icon
          }, {
            default: function _default() {
              return [typeof _content === 'function' ? _content() : _content];
            }
          });
        },
        placement: 'top',
        // @ts-ignore
        class: classNames(type && "".concat(noticePrefixCls, "-").concat(type), hashId, className),
        onClose: function onClose() {
          _onClose === null || _onClose === void 0 ? void 0 : _onClose();
          resolve();
        }
      }));
      // Return close function
      return function () {
        close(mergedKey);
      };
    });
  };
  // >>> destroy
  var destroy = function destroy(key) {
    var _a;
    if (key !== undefined) {
      close(key);
    } else {
      (_a = holderRef.value) === null || _a === void 0 ? void 0 : _a.destroy();
    }
  };
  var wrapAPI = {
    open: open,
    destroy: destroy
  };
  var keys = ['info', 'success', 'warning', 'error', 'loading'];
  keys.forEach(function (type) {
    var typeOpen = function typeOpen(jointContent, duration, onClose) {
      var config;
      if (jointContent && _typeof(jointContent) === 'object' && 'content' in jointContent) {
        config = jointContent;
      } else {
        config = {
          content: jointContent
        };
      }
      // Params
      var mergedDuration;
      var mergedOnClose;
      if (typeof duration === 'function') {
        mergedOnClose = duration;
      } else {
        mergedDuration = duration;
        mergedOnClose = onClose;
      }
      var mergedConfig = _extends(_extends({
        onClose: mergedOnClose,
        duration: mergedDuration
      }, config), {
        type: type
      });
      return open(mergedConfig);
    };
    wrapAPI[type] = typeOpen;
  });
  // ============================== Return ===============================
  return [wrapAPI, function () {
    return _createVNode(Holder, _objectSpread(_objectSpread({
      "key": holderKey
    }, messageConfig), {}, {
      "ref": holderRef
    }), null);
  }];
}
export default function useMessage(messageConfig) {
  return useInternalMessage(messageConfig);
}
