// ant-design-vue/lib/notification/style/placement.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _cssinjs = require("../../_util/cssinjs");
var genNotificationPlacementStyle = function genNotificationPlacementStyle(token) {
  var _ref;
  var componentCls = token.componentCls,
    width = token.width,
    notificationMarginEdge = token.notificationMarginEdge;
  var notificationTopFadeIn = new _cssinjs.Keyframes('antNotificationTopFadeIn', {
    '0%': {
      marginTop: '-100%',
      opacity: 0
    },
    '100%': {
      marginTop: 0,
      opacity: 1
    }
  });
  var notificationBottomFadeIn = new _cssinjs.Keyframes('antNotificationBottomFadeIn', {
    '0%': {
      marginBottom: '-100%',
      opacity: 0
    },
    '100%': {
      marginBottom: 0,
      opacity: 1
    }
  });
  var notificationLeftFadeIn = new _cssinjs.Keyframes('antNotificationLeftFadeIn', {
    '0%': {
      right: {
        _skip_check_: true,
        value: width
      },
      opacity: 0
    },
    '100%': {
      right: {
        _skip_check_: true,
        value: 0
      },
      opacity: 1
    }
  });
  return _ref = {}, (0, _defineProperty2.default)(_ref, "&".concat(componentCls, "-top, &").concat(componentCls, "-bottom"), {
    marginInline: 0
  }), (0, _defineProperty2.default)(_ref, "&".concat(componentCls, "-top"), (0, _defineProperty2.default)({}, "".concat(componentCls, "-fade-enter").concat(componentCls, "-fade-enter-active, ").concat(componentCls, "-fade-appear").concat(componentCls, "-fade-appear-active"), {
    animationName: notificationTopFadeIn
  })), (0, _defineProperty2.default)(_ref, "&".concat(componentCls, "-bottom"), (0, _defineProperty2.default)({}, "".concat(componentCls, "-fade-enter").concat(componentCls, "-fade-enter-active, ").concat(componentCls, "-fade-appear").concat(componentCls, "-fade-appear-active"), {
    animationName: notificationBottomFadeIn
  })), (0, _defineProperty2.default)(_ref, "&".concat(componentCls, "-topLeft, &").concat(componentCls, "-bottomLeft"), (0, _defineProperty2.default)({
    marginInlineEnd: 0,
    marginInlineStart: notificationMarginEdge
  }, "".concat(componentCls, "-fade-enter").concat(componentCls, "-fade-enter-active, ").concat(componentCls, "-fade-appear").concat(componentCls, "-fade-appear-active"), {
    animationName: notificationLeftFadeIn
  })), _ref;
};
var _default = genNotificationPlacementStyle;
exports.default = _default;
