// ant-design-vue/es/steps/style/label-placement.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
var genStepsLabelPlacementStyle = function genStepsLabelPlacementStyle(token) {
  var _$concat;
  var componentCls = token.componentCls,
    stepsIconSize = token.stepsIconSize,
    lineHeight = token.lineHeight,
    stepsSmallIconSize = token.stepsSmallIconSize;
  return _defineProperty({}, "&".concat(componentCls, "-label-vertical"), (_$concat = {}, _defineProperty(_$concat, "".concat(componentCls, "-item"), {
    overflow: 'visible',
    '&-tail': {
      marginInlineStart: stepsIconSize / 2 + token.controlHeightLG,
      padding: "".concat(token.paddingXXS, "px ").concat(token.paddingLG, "px")
    },
    '&-content': {
      display: 'block',
      width: (stepsIconSize / 2 + token.controlHeightLG) * 2,
      marginTop: token.marginSM,
      textAlign: 'center'
    },
    '&-icon': {
      display: 'inline-block',
      marginInlineStart: token.controlHeightLG
    },
    '&-title': {
      paddingInlineEnd: 0,
      paddingInlineStart: 0,
      '&::after': {
        display: 'none'
      }
    },
    '&-subtitle': {
      display: 'block',
      marginBottom: token.marginXXS,
      marginInlineStart: 0,
      lineHeight: lineHeight
    }
  }), _defineProperty(_$concat, "&".concat(componentCls, "-small:not(").concat(componentCls, "-dot)"), _defineProperty({}, "".concat(componentCls, "-item"), {
    '&-icon': {
      marginInlineStart: token.controlHeightLG + (stepsIconSize - stepsSmallIconSize) / 2
    }
  })), _$concat));
};
export default genStepsLabelPlacementStyle;
