// ant-design-vue/es/steps/style/small.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
var genStepsSmallStyle = function genStepsSmallStyle(token) {
  var _$concat;
  var componentCls = token.componentCls,
    stepsSmallIconSize = token.stepsSmallIconSize,
    fontSizeSM = token.fontSizeSM,
    fontSize = token.fontSize,
    colorTextDescription = token.colorTextDescription;
  return _defineProperty({}, "&".concat(componentCls, "-small"), (_$concat = {}, _defineProperty(_$concat, "&".concat(componentCls, "-horizontal:not(").concat(componentCls, "-label-vertical) ").concat(componentCls, "-item"), {
    paddingInlineStart: token.paddingSM,
    '&:first-child': {
      paddingInlineStart: 0
    }
  }), _defineProperty(_$concat, "".concat(componentCls, "-item-icon"), {
    width: stepsSmallIconSize,
    height: stepsSmallIconSize,
    // margin: stepsSmallIconMargin,
    marginTop: 0,
    marginBottom: 0,
    marginInline: "0 ".concat(token.marginXS, "px"),
    fontSize: fontSizeSM,
    lineHeight: "".concat(stepsSmallIconSize, "px"),
    textAlign: 'center',
    borderRadius: stepsSmallIconSize
  }), _defineProperty(_$concat, "".concat(componentCls, "-item-title"), {
    paddingInlineEnd: token.paddingSM,
    fontSize: fontSize,
    lineHeight: "".concat(stepsSmallIconSize, "px"),
    '&::after': {
      top: stepsSmallIconSize / 2
    }
  }), _defineProperty(_$concat, "".concat(componentCls, "-item-description"), {
    color: colorTextDescription,
    fontSize: fontSize
  }), _defineProperty(_$concat, "".concat(componentCls, "-item-tail"), {
    top: stepsSmallIconSize / 2 - token.paddingXXS
  }), _defineProperty(_$concat, "".concat(componentCls, "-item-custom ").concat(componentCls, "-item-icon"), _defineProperty({
    width: 'inherit',
    height: 'inherit',
    lineHeight: 'inherit',
    background: 'none',
    border: 0,
    borderRadius: 0
  }, "> ".concat(componentCls, "-icon"), {
    fontSize: stepsSmallIconSize,
    lineHeight: "".concat(stepsSmallIconSize, "px"),
    transform: 'none'
  })), _$concat));
};
export default genStepsSmallStyle;
