// ant-design-vue/lib/float-button/FloatButtonContent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _vue = require("vue");
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _FileTextOutlined = _interopRequireDefault(require("@ant-design/icons-vue/lib/icons/FileTextOutlined"));
var _interface = require("./interface");
var _propsUtil = require("../_util/props-util");
var FloatButtonContent = (0, _vue.defineComponent)({
  compatConfig: {
    MODE: 3
  },
  name: 'AFloatButtonContent',
  inheritAttrs: false,
  props: (0, _interface.floatButtonContentProps)(),
  setup: function setup(props, _ref) {
    var attrs = _ref.attrs,
      slots = _ref.slots;
    return function () {
      var _a;
      var prefixCls = props.prefixCls;
      var description = (0, _propsUtil.filterEmpty)((_a = slots.description) === null || _a === void 0 ? void 0 : _a.call(slots));
      return (0, _vue.createVNode)("div", (0, _objectSpread2.default)((0, _objectSpread2.default)({}, attrs), {}, {
        "class": [attrs.class, "".concat(prefixCls, "-content")]
      }), [slots.icon || description.length ? (0, _vue.createVNode)(_vue.Fragment, null, [slots.icon && (0, _vue.createVNode)("div", {
        "class": "".concat(prefixCls, "-icon")
      }, [slots.icon()]), description.length ? (0, _vue.createVNode)("div", {
        "class": "".concat(prefixCls, "-description")
      }, [description]) : null]) : (0, _vue.createVNode)("div", {
        "class": "".concat(prefixCls, "-icon")
      }, [(0, _vue.createVNode)(_FileTextOutlined.default, null, null)])]);
    };
  }
});
var _default = FloatButtonContent;
exports.default = _default;
