// ant-design-vue/es/upload/style/motion.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import { Keyframes } from '../../_util/cssinjs';
var uploadAnimateInlineIn = new Keyframes('uploadAnimateInlineIn', {
  from: {
    width: 0,
    height: 0,
    margin: 0,
    padding: 0,
    opacity: 0
  }
});
var uploadAnimateInlineOut = new Keyframes('uploadAnimateInlineOut', {
  to: {
    width: 0,
    height: 0,
    margin: 0,
    padding: 0,
    opacity: 0
  }
});
// =========================== Motion ===========================
var genMotionStyle = function genMotionStyle(token) {
  var _$concat;
  var componentCls = token.componentCls;
  var inlineCls = "".concat(componentCls, "-animate-inline");
  return [_defineProperty({}, "".concat(componentCls, "-wrapper"), (_$concat = {}, _defineProperty(_$concat, "".concat(inlineCls, "-appear, ").concat(inlineCls, "-enter, ").concat(inlineCls, "-leave"), {
    animationDuration: token.motionDurationSlow,
    animationTimingFunction: token.motionEaseInOutCirc,
    animationFillMode: 'forwards'
  }), _defineProperty(_$concat, "".concat(inlineCls, "-appear, ").concat(inlineCls, "-enter"), {
    animationName: uploadAnimateInlineIn
  }), _defineProperty(_$concat, "".concat(inlineCls, "-leave"), {
    animationName: uploadAnimateInlineOut
  }), _$concat)), uploadAnimateInlineIn, uploadAnimateInlineOut];
};
export default genMotionStyle;
