// ant-design-vue/es/config-provider/style/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _extends from "@babel/runtime/helpers/esm/extends";
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import { useStyleRegister } from '../../_util/cssinjs';
import { resetIcon } from '../../style';
import { useToken } from '../../theme/internal';
import { computed } from 'vue';
var useStyle = function useStyle(iconPrefixCls) {
  var _useToken = useToken(),
    _useToken2 = _slicedToArray(_useToken, 2),
    theme = _useToken2[0],
    token = _useToken2[1];
  // Generate style for icons
  return useStyleRegister(computed(function () {
    return {
      theme: theme.value,
      token: token.value,
      hashId: '',
      path: ['ant-design-icons', iconPrefixCls.value]
    };
  }), function () {
    return [_defineProperty({}, ".".concat(iconPrefixCls.value), _extends(_extends({}, resetIcon()), _defineProperty({}, ".".concat(iconPrefixCls.value, " .").concat(iconPrefixCls.value, "-icon"), {
      display: 'block'
    })))];
  });
};
export default useStyle;
