// ant-design-vue/es/table/style/resize.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
var genResizeStyle = function genResizeStyle(token) {
  var _$concat$concat, _$concat$concat2, _ref;
  var componentCls = token.componentCls;
  return _ref = {}, _defineProperty(_ref, "".concat(componentCls, "-wrapper ").concat(componentCls, "-resize-handle"), (_$concat$concat = {
    position: 'absolute',
    top: 0,
    height: '100% !important',
    bottom: 0,
    left: ' auto !important',
    right: ' -8px',
    cursor: 'col-resize',
    touchAction: 'none',
    userSelect: 'auto',
    width: '16px',
    zIndex: 1
  }, _defineProperty(_$concat$concat, "&-line", {
    display: 'block',
    width: '1px',
    marginLeft: '7px',
    height: '100% !important',
    backgroundColor: token.colorPrimary,
    opacity: 0
  }), _defineProperty(_$concat$concat, "&:hover &-line", {
    opacity: 1
  }), _$concat$concat)), _defineProperty(_ref, "".concat(componentCls, "-wrapper  ").concat(componentCls, "-resize-handle.dragging"), (_$concat$concat2 = {
    overflow: 'hidden'
  }, _defineProperty(_$concat$concat2, "".concat(componentCls, "-resize-handle-line"), {
    opacity: 1
  }), _defineProperty(_$concat$concat2, "&:before", {
    position: 'absolute',
    top: 0,
    bottom: 0,
    content: '" "',
    width: '200vw',
    transform: 'translateX(-50%)',
    opacity: 0
  }), _$concat$concat2)), _ref;
};
export default genResizeStyle;
