// ant-design-vue/es/table/style/selection.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
var genSelectionStyle = function genSelectionStyle(token) {
  var _$concat2;
  var componentCls = token.componentCls,
    antCls = token.antCls,
    iconCls = token.iconCls,
    fontSizeIcon = token.fontSizeIcon,
    paddingXS = token.paddingXS,
    tableHeaderIconColor = token.tableHeaderIconColor,
    tableHeaderIconColorHover = token.tableHeaderIconColorHover;
  return _defineProperty({}, "".concat(componentCls, "-wrapper"), (_$concat2 = {}, _defineProperty(_$concat2, "".concat(componentCls, "-selection-col"), {
    width: token.tableSelectionColumnWidth
  }), _defineProperty(_$concat2, "".concat(componentCls, "-bordered ").concat(componentCls, "-selection-col"), {
    width: token.tableSelectionColumnWidth + paddingXS * 2
  }), _defineProperty(_$concat2, "\n        table tr th".concat(componentCls, "-selection-column,\n        table tr td").concat(componentCls, "-selection-column\n      "), _defineProperty({
    paddingInlineEnd: token.paddingXS,
    paddingInlineStart: token.paddingXS,
    textAlign: 'center'
  }, "".concat(antCls, "-radio-wrapper"), {
    marginInlineEnd: 0
  })), _defineProperty(_$concat2, "table tr th".concat(componentCls, "-selection-column").concat(componentCls, "-cell-fix-left"), {
    zIndex: token.zIndexTableFixed + 1
  }), _defineProperty(_$concat2, "table tr th".concat(componentCls, "-selection-column::after"), {
    backgroundColor: 'transparent !important'
  }), _defineProperty(_$concat2, "".concat(componentCls, "-selection"), {
    position: 'relative',
    display: 'inline-flex',
    flexDirection: 'column'
  }), _defineProperty(_$concat2, "".concat(componentCls, "-selection-extra"), _defineProperty({
    position: 'absolute',
    top: 0,
    zIndex: 1,
    cursor: 'pointer',
    transition: "all ".concat(token.motionDurationSlow),
    marginInlineStart: '100%',
    paddingInlineStart: "".concat(token.tablePaddingHorizontal / 4, "px")
  }, iconCls, {
    color: tableHeaderIconColor,
    fontSize: fontSizeIcon,
    verticalAlign: 'baseline',
    '&:hover': {
      color: tableHeaderIconColorHover
    }
  })), _$concat2));
};
export default genSelectionStyle;
