// ant-design-vue/lib/notification/useNotification.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = useNotification;
exports.useInternalNotification = useInternalNotification;
var _vue = require("vue");
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _vcNotification = require("../vc-notification");
var _style = _interopRequireDefault(require("./style"));
var _PurePanel = require("./PurePanel");
var _util = require("./util");
var _useConfigInject2 = _interopRequireDefault(require("../config-provider/hooks/useConfigInject"));
var _classNames2 = _interopRequireDefault(require("../_util/classNames"));
var __rest = void 0 && (void 0).__rest || function (s, e) {
  var t = {};
  for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};
var DEFAULT_OFFSET = 24;
var DEFAULT_DURATION = 4.5;
var Holder = (0, _vue.defineComponent)({
  name: 'Holder',
  inheritAttrs: false,
  props: ['prefixCls', 'class', 'type', 'icon', 'content', 'onAllRemoved'],
  setup: function setup(props, _ref) {
    var expose = _ref.expose;
    var _useConfigInject = (0, _useConfigInject2.default)('notification', props),
      getPrefixCls = _useConfigInject.getPrefixCls,
      getPopupContainer = _useConfigInject.getPopupContainer;
    var prefixCls = (0, _vue.computed)(function () {
      return props.prefixCls || getPrefixCls('notification');
    });
    // =============================== Style ===============================
    var getStyles = function getStyles(placement) {
      var _a, _b;
      return (0, _util.getPlacementStyle)(placement, (_a = props.top) !== null && _a !== void 0 ? _a : DEFAULT_OFFSET, (_b = props.bottom) !== null && _b !== void 0 ? _b : DEFAULT_OFFSET);
    };
    // Style
    var _useStyle = (0, _style.default)(prefixCls),
      _useStyle2 = (0, _slicedToArray2.default)(_useStyle, 2),
      hashId = _useStyle2[1];
    var getClassName = function getClassName() {
      return (0, _classNames2.default)(hashId.value, (0, _defineProperty2.default)({}, "".concat(prefixCls.value, "-rtl"), props.rtl));
    };
    // ============================== Motion ===============================
    var getNotificationMotion = function getNotificationMotion() {
      return (0, _util.getMotion)(prefixCls.value);
    };
    // ============================== Origin ===============================
    var _useVcNotification = (0, _vcNotification.useNotification)({
        prefixCls: prefixCls.value,
        getStyles: getStyles,
        getClassName: getClassName,
        motion: getNotificationMotion,
        closable: true,
        closeIcon: (0, _PurePanel.getCloseIcon)(prefixCls.value),
        duration: DEFAULT_DURATION,
        getContainer: function getContainer() {
          var _a, _b;
          return ((_a = props.getPopupContainer) === null || _a === void 0 ? void 0 : _a.call(props)) || ((_b = getPopupContainer.value) === null || _b === void 0 ? void 0 : _b.call(getPopupContainer)) || document.body;
        },
        maxCount: props.maxCount,
        hashId: hashId.value,
        onAllRemoved: props.onAllRemoved
      }),
      _useVcNotification2 = (0, _slicedToArray2.default)(_useVcNotification, 2),
      api = _useVcNotification2[0],
      holder = _useVcNotification2[1];
    // ================================ Ref ================================
    expose((0, _extends2.default)((0, _extends2.default)({}, api), {
      prefixCls: prefixCls.value,
      hashId: hashId
    }));
    return holder;
  }
});
// ==============================================================================
// ==                                   Hook                                   ==
// ==============================================================================
function useInternalNotification(notificationConfig) {
  var holderRef = (0, _vue.shallowRef)(null);
  var holderKey = Symbol('notificationHolderKey');
  // ================================ API ================================
  // Wrap with notification content
  // >>> Open
  var open = function open(config) {
    if (!holderRef.value) {
      return;
    }
    var _holderRef$value = holderRef.value,
      originOpen = _holderRef$value.open,
      prefixCls = _holderRef$value.prefixCls,
      hashId = _holderRef$value.hashId;
    var noticePrefixCls = "".concat(prefixCls, "-notice");
    var message = config.message,
      description = config.description,
      icon = config.icon,
      type = config.type,
      btn = config.btn,
      className = config.class,
      restConfig = __rest(config, ["message", "description", "icon", "type", "btn", "class"]);
    return originOpen((0, _extends2.default)((0, _extends2.default)({
      placement: 'topRight'
    }, restConfig), {
      content: function content() {
        return (0, _vue.createVNode)(_PurePanel.PureContent, {
          "prefixCls": noticePrefixCls,
          "icon": typeof icon === 'function' ? icon() : icon,
          "type": type,
          "message": typeof message === 'function' ? message() : message,
          "description": typeof description === 'function' ? description() : description,
          "btn": typeof btn === 'function' ? btn() : btn
        }, null);
      },
      // @ts-ignore
      class: (0, _classNames2.default)(type && "".concat(noticePrefixCls, "-").concat(type), hashId, className)
    }));
  };
  // >>> destroy
  var destroy = function destroy(key) {
    var _a, _b;
    if (key !== undefined) {
      (_a = holderRef.value) === null || _a === void 0 ? void 0 : _a.close(key);
    } else {
      (_b = holderRef.value) === null || _b === void 0 ? void 0 : _b.destroy();
    }
  };
  var wrapAPI = {
    open: open,
    destroy: destroy
  };
  var keys = ['success', 'info', 'warning', 'error'];
  keys.forEach(function (type) {
    wrapAPI[type] = function (config) {
      return open((0, _extends2.default)((0, _extends2.default)({}, config), {
        type: type
      }));
    };
  });
  // ============================== Return ===============================
  return [wrapAPI, function () {
    return (0, _vue.createVNode)(Holder, (0, _objectSpread2.default)((0, _objectSpread2.default)({
      "key": holderKey
    }, notificationConfig), {}, {
      "ref": holderRef
    }), null);
  }];
}
function useNotification(notificationConfig) {
  return useInternalNotification(notificationConfig);
}
