// ant-design-vue/lib/config-provider/style/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _extends3 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _cssinjs = require("../../_util/cssinjs");
var _style = require("../../style");
var _internal = require("../../theme/internal");
var _vue = require("vue");
var useStyle = function useStyle(iconPrefixCls) {
  var _useToken = (0, _internal.useToken)(),
    _useToken2 = (0, _slicedToArray2.default)(_useToken, 2),
    theme = _useToken2[0],
    token = _useToken2[1];
  // Generate style for icons
  return (0, _cssinjs.useStyleRegister)((0, _vue.computed)(function () {
    return {
      theme: theme.value,
      token: token.value,
      hashId: '',
      path: ['ant-design-icons', iconPrefixCls.value]
    };
  }), function () {
    return [(0, _defineProperty2.default)({}, ".".concat(iconPrefixCls.value), (0, _extends3.default)((0, _extends3.default)({}, (0, _style.resetIcon)()), (0, _defineProperty2.default)({}, ".".concat(iconPrefixCls.value, " .").concat(iconPrefixCls.value, "-icon"), {
      display: 'block'
    })))];
  });
};
var _default = useStyle;
exports.default = _default;
