// ant-design-vue/lib/steps/style/nav.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _style = require("../../style");
var genStepsNavStyle = function genStepsNavStyle(token) {
  var _container, _$concat2, _$concat3, _$concat4, _ref;
  var componentCls = token.componentCls,
    stepsNavContentMaxWidth = token.stepsNavContentMaxWidth,
    stepsNavArrowColor = token.stepsNavArrowColor,
    stepsNavActiveColor = token.stepsNavActiveColor,
    motionDurationSlow = token.motionDurationSlow;
  return _ref = {}, (0, _defineProperty2.default)(_ref, "&".concat(componentCls, "-navigation"), (_$concat3 = {
    paddingTop: token.paddingSM
  }, (0, _defineProperty2.default)(_$concat3, "&".concat(componentCls, "-small"), (0, _defineProperty2.default)({}, "".concat(componentCls, "-item"), {
    '&-container': {
      marginInlineStart: -token.marginSM
    }
  })), (0, _defineProperty2.default)(_$concat3, "".concat(componentCls, "-item"), (_$concat2 = {
    overflow: 'visible',
    textAlign: 'center',
    '&-container': (_container = {
      display: 'inline-block',
      height: '100%',
      marginInlineStart: -token.margin,
      paddingBottom: token.paddingSM,
      textAlign: 'start',
      transition: "opacity ".concat(motionDurationSlow)
    }, (0, _defineProperty2.default)(_container, "".concat(componentCls, "-item-content"), {
      maxWidth: stepsNavContentMaxWidth
    }), (0, _defineProperty2.default)(_container, "".concat(componentCls, "-item-title"), (0, _extends2.default)((0, _extends2.default)({
      maxWidth: '100%',
      paddingInlineEnd: 0
    }, _style.textEllipsis), {
      '&::after': {
        display: 'none'
      }
    })), _container)
  }, (0, _defineProperty2.default)(_$concat2, "&:not(".concat(componentCls, "-item-active)"), (0, _defineProperty2.default)({}, "".concat(componentCls, "-item-container[role='button']"), {
    cursor: 'pointer',
    '&:hover': {
      opacity: 0.85
    }
  })), (0, _defineProperty2.default)(_$concat2, '&:last-child', {
    flex: 1,
    '&::after': {
      display: 'none'
    }
  }), (0, _defineProperty2.default)(_$concat2, '&::after', {
    position: 'absolute',
    top: "calc(50% - ".concat(token.paddingSM / 2, "px)"),
    insetInlineStart: '100%',
    display: 'inline-block',
    width: token.fontSizeIcon,
    height: token.fontSizeIcon,
    borderTop: "".concat(token.lineWidth, "px ").concat(token.lineType, " ").concat(stepsNavArrowColor),
    borderBottom: 'none',
    borderInlineStart: 'none',
    borderInlineEnd: "".concat(token.lineWidth, "px ").concat(token.lineType, " ").concat(stepsNavArrowColor),
    transform: 'translateY(-50%) translateX(-50%) rotate(45deg)',
    content: '""'
  }), (0, _defineProperty2.default)(_$concat2, '&::before', {
    position: 'absolute',
    bottom: 0,
    insetInlineStart: '50%',
    display: 'inline-block',
    width: 0,
    height: token.lineWidthBold,
    backgroundColor: stepsNavActiveColor,
    transition: "width ".concat(motionDurationSlow, ", inset-inline-start ").concat(motionDurationSlow),
    transitionTimingFunction: 'ease-out',
    content: '""'
  }), _$concat2)), (0, _defineProperty2.default)(_$concat3, "".concat(componentCls, "-item").concat(componentCls, "-item-active::before"), {
    insetInlineStart: 0,
    width: '100%'
  }), _$concat3)), (0, _defineProperty2.default)(_ref, "&".concat(componentCls, "-navigation").concat(componentCls, "-vertical"), (0, _defineProperty2.default)({}, "> ".concat(componentCls, "-item"), (_$concat4 = {
    marginInlineEnd: 0,
    '&::before': {
      display: 'none'
    }
  }, (0, _defineProperty2.default)(_$concat4, "&".concat(componentCls, "-item-active::before"), {
    top: 0,
    insetInlineEnd: 0,
    insetInlineStart: 'unset',
    display: 'block',
    width: token.lineWidth * 3,
    height: "calc(100% - ".concat(token.marginLG, "px)")
  }), (0, _defineProperty2.default)(_$concat4, '&::after', {
    position: 'relative',
    insetInlineStart: '50%',
    display: 'block',
    width: token.controlHeight * 0.25,
    height: token.controlHeight * 0.25,
    marginBottom: token.marginXS,
    textAlign: 'center',
    transform: 'translateY(-50%) translateX(-50%) rotate(135deg)'
  }), (0, _defineProperty2.default)(_$concat4, "> ".concat(componentCls, "-item-container > ").concat(componentCls, "-item-tail"), {
    visibility: 'hidden'
  }), _$concat4))), (0, _defineProperty2.default)(_ref, "&".concat(componentCls, "-navigation").concat(componentCls, "-horizontal"), (0, _defineProperty2.default)({}, "> ".concat(componentCls, "-item > ").concat(componentCls, "-item-container > ").concat(componentCls, "-item-tail"), {
    visibility: 'hidden'
  })), _ref;
};
var _default = genStepsNavStyle;
exports.default = _default;
