// ant-design-vue/lib/steps/style/progress.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var genStepsProgressStyle = function genStepsProgressStyle(token) {
  var _$concat5;
  var antCls = token.antCls,
    componentCls = token.componentCls;
  return (0, _defineProperty2.default)({}, "&".concat(componentCls, "-with-progress"), (_$concat5 = {}, (0, _defineProperty2.default)(_$concat5, "".concat(componentCls, "-item"), (0, _defineProperty2.default)({
    paddingTop: token.paddingXXS
  }, "&-process ".concat(componentCls, "-item-container ").concat(componentCls, "-item-icon ").concat(componentCls, "-icon"), {
    color: token.processIconColor
  })), (0, _defineProperty2.default)(_$concat5, "&".concat(componentCls, "-vertical > ").concat(componentCls, "-item "), (0, _defineProperty2.default)({
    paddingInlineStart: token.paddingXXS
  }, "> ".concat(componentCls, "-item-container > ").concat(componentCls, "-item-tail"), {
    top: token.marginXXS,
    insetInlineStart: token.stepsIconSize / 2 - token.lineWidth + token.paddingXXS
  })), (0, _defineProperty2.default)(_$concat5, "&, &".concat(componentCls, "-small"), (0, _defineProperty2.default)({}, "&".concat(componentCls, "-horizontal ").concat(componentCls, "-item:first-child"), {
    paddingBottom: token.paddingXXS,
    paddingInlineStart: token.paddingXXS
  })), (0, _defineProperty2.default)(_$concat5, "&".concat(componentCls, "-small").concat(componentCls, "-vertical > ").concat(componentCls, "-item > ").concat(componentCls, "-item-container > ").concat(componentCls, "-item-tail"), {
    insetInlineStart: token.stepsSmallIconSize / 2 - token.lineWidth + token.paddingXXS
  }), (0, _defineProperty2.default)(_$concat5, "&".concat(componentCls, "-label-vertical"), (0, _defineProperty2.default)({}, "".concat(componentCls, "-item ").concat(componentCls, "-item-tail"), {
    top: token.margin - 2 * token.lineWidth
  })), (0, _defineProperty2.default)(_$concat5, "".concat(componentCls, "-item-icon"), (0, _defineProperty2.default)({
    position: 'relative'
  }, "".concat(antCls, "-progress"), {
    position: 'absolute',
    insetBlockStart: (token.stepsIconSize - token.stepsProgressSize - token.lineWidth * 2) / 2,
    insetInlineStart: (token.stepsIconSize - token.stepsProgressSize - token.lineWidth * 2) / 2
  })), _$concat5));
};
var _default = genStepsProgressStyle;
exports.default = _default;
