// ant-design-vue/lib/steps/style/small.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var genStepsSmallStyle = function genStepsSmallStyle(token) {
  var _$concat;
  var componentCls = token.componentCls,
    stepsSmallIconSize = token.stepsSmallIconSize,
    fontSizeSM = token.fontSizeSM,
    fontSize = token.fontSize,
    colorTextDescription = token.colorTextDescription;
  return (0, _defineProperty2.default)({}, "&".concat(componentCls, "-small"), (_$concat = {}, (0, _defineProperty2.default)(_$concat, "&".concat(componentCls, "-horizontal:not(").concat(componentCls, "-label-vertical) ").concat(componentCls, "-item"), {
    paddingInlineStart: token.paddingSM,
    '&:first-child': {
      paddingInlineStart: 0
    }
  }), (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-item-icon"), {
    width: stepsSmallIconSize,
    height: stepsSmallIconSize,
    // margin: stepsSmallIconMargin,
    marginTop: 0,
    marginBottom: 0,
    marginInline: "0 ".concat(token.marginXS, "px"),
    fontSize: fontSizeSM,
    lineHeight: "".concat(stepsSmallIconSize, "px"),
    textAlign: 'center',
    borderRadius: stepsSmallIconSize
  }), (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-item-title"), {
    paddingInlineEnd: token.paddingSM,
    fontSize: fontSize,
    lineHeight: "".concat(stepsSmallIconSize, "px"),
    '&::after': {
      top: stepsSmallIconSize / 2
    }
  }), (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-item-description"), {
    color: colorTextDescription,
    fontSize: fontSize
  }), (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-item-tail"), {
    top: stepsSmallIconSize / 2 - token.paddingXXS
  }), (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-item-custom ").concat(componentCls, "-item-icon"), (0, _defineProperty2.default)({
    width: 'inherit',
    height: 'inherit',
    lineHeight: 'inherit',
    background: 'none',
    border: 0,
    borderRadius: 0
  }, "> ".concat(componentCls, "-icon"), {
    fontSize: stepsSmallIconSize,
    lineHeight: "".concat(stepsSmallIconSize, "px"),
    transform: 'none'
  })), _$concat));
};
var _default = genStepsSmallStyle;
exports.default = _default;
