// ant-design-vue/lib/steps/style/vertical.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var genStepsVerticalStyle = function genStepsVerticalStyle(token) {
  var _$concat, _$concat$concat, _$concat2;
  var componentCls = token.componentCls,
    stepsSmallIconSize = token.stepsSmallIconSize,
    stepsIconSize = token.stepsIconSize;
  return (0, _defineProperty2.default)({}, "&".concat(componentCls, "-vertical"), (_$concat2 = {
    display: 'flex',
    flexDirection: 'column'
  }, (0, _defineProperty2.default)(_$concat2, "> ".concat(componentCls, "-item"), (_$concat = {
    display: 'block',
    flex: '1 0 auto',
    paddingInlineStart: 0,
    overflow: 'visible'
  }, (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-item-icon"), {
    float: 'left',
    marginInlineEnd: token.margin
  }), (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-item-content"), {
    display: 'block',
    minHeight: token.controlHeight * 1.5,
    overflow: 'hidden'
  }), (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-item-title"), {
    lineHeight: "".concat(stepsIconSize, "px")
  }), (0, _defineProperty2.default)(_$concat, "".concat(componentCls, "-item-description"), {
    paddingBottom: token.paddingSM
  }), _$concat)), (0, _defineProperty2.default)(_$concat2, "> ".concat(componentCls, "-item > ").concat(componentCls, "-item-container > ").concat(componentCls, "-item-tail"), {
    position: 'absolute',
    top: 0,
    insetInlineStart: token.stepsIconSize / 2 - token.lineWidth,
    width: token.lineWidth,
    height: '100%',
    padding: "".concat(stepsIconSize + token.marginXXS * 1.5, "px 0 ").concat(token.marginXXS * 1.5, "px"),
    '&::after': {
      width: token.lineWidth,
      height: '100%'
    }
  }), (0, _defineProperty2.default)(_$concat2, "> ".concat(componentCls, "-item:not(:last-child) > ").concat(componentCls, "-item-container > ").concat(componentCls, "-item-tail"), {
    display: 'block'
  }), (0, _defineProperty2.default)(_$concat2, " > ".concat(componentCls, "-item > ").concat(componentCls, "-item-container > ").concat(componentCls, "-item-content > ").concat(componentCls, "-item-title"), {
    '&::after': {
      display: 'none'
    }
  }), (0, _defineProperty2.default)(_$concat2, "&".concat(componentCls, "-small ").concat(componentCls, "-item-container"), (_$concat$concat = {}, (0, _defineProperty2.default)(_$concat$concat, "".concat(componentCls, "-item-tail"), {
    position: 'absolute',
    top: 0,
    insetInlineStart: token.stepsSmallIconSize / 2 - token.lineWidth,
    padding: "".concat(stepsSmallIconSize + token.marginXXS * 1.5, "px 0 ").concat(token.marginXXS * 1.5, "px")
  }), (0, _defineProperty2.default)(_$concat$concat, "".concat(componentCls, "-item-title"), {
    lineHeight: "".concat(stepsSmallIconSize, "px")
  }), _$concat$concat)), _$concat2));
};
var _default = genStepsVerticalStyle;
exports.default = _default;
